{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# QC on chromosome X: LBD (path + clin) vs. Controls (DementiaSeq, Wellderly, LNG) \n",
    "\n",
    "**Start date:** 07-27-2020\n",
    "\n",
    "**Updated date:** 2024-01-10\n",
    "\n",
    "**Author(s):** Ruth Chia\n",
    "\n",
    "**Working directory on biowulf:** `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/vf/users/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What to do\n",
    "1. Run chrX specific QC (variant level) separately on males and females; \n",
    "    - missingness by case-control (keep vars with  P > 1e-4)\n",
    "    - missingness by haplotype case-control  (keep vars with  P > 1e-4)\n",
    "    - keep vars with geno > 98% (i.e. --geno 0.02)\n",
    "2. Then merge ; then ran another round of QC on merged data with additional parameters below   \n",
    "3. Run XWAS\n",
    "\n",
    "\n",
    "## QC parameters specific for chr X\n",
    "1. Remove SNPs with significantly different MAF between male and female samples in the control group (use xwas --diff-x)\n",
    "2. Remove SNPs with significantly different missingness rates between male and female controls (--test-missing ; code male = 1/\"control\" and female = 2/\"case\") or use --xwas --missdiff-x\n",
    "3. Remove SNPs in the pseudoautosomal regions (PARs). \n",
    "4. Remove SNPs with HWE (female controls) midp < 1e-4\n",
    "\n",
    "\n",
    "\n",
    "## What files I need\n",
    "1. UnQC-ed chrX: `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/neurod.freeze9.LBD.FTD.ALLcontrols4K.chrX.pgen/pvar/psam`\n",
    "\n",
    "2. Make list of QC-ed LBD samples and controls from: `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/CLEAN.UNRELATED/FILTERED.LBD.controls.UNRELATED_chrX.psam`\n",
    "\n",
    "\n",
    "## Relevant papers\n",
    "- `https://onlinelibrary.wiley.com/doi/10.1002/gepi.21782`\n",
    "- `https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4567842/`\n",
    "- `https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7132553/`\n",
    "- `https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6781167/`\n",
    "- `https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6646348/`\n",
    "- `https://pubmed.ncbi.nlm.nih.gov/25479423/`\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create unQC-ed chrX containing LBD samples + controls for chr X specific QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load plink/2.0-dev-20191128\n",
    "\n",
    "DATA=\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020\"\n",
    "\n",
    "# Create list of samples to keep\n",
    "awk '{print $1,$2}' OFS=\" \" $DATA/LBD/CLEAN.UNRELATED/FILTERED.LBD.controls.UNRELATED_chrX.psam > SampleList.LBD.controls.UNRELATED.FIDspaceIID.txt\n",
    "\n",
    "# Make PAR region exclusion file for removal later\n",
    "echo \"23 10001 2781479 PAR1\" > scripts/PAR_hg38_region.txt\n",
    "echo \"23 155701383 156030895 PAR2\" >> scripts/PAR_hg38_region.txt\n",
    "\n",
    "# Subset chr X\n",
    "plink \\\n",
    "--pfile $DATA/neurod.freeze9.LBD.FTD.ALLcontrols4K.chrX \\\n",
    "--keep SampleList.LBD.controls.UNRELATED.FIDspaceIID.txt \\\n",
    "--set-hh-missing \\\n",
    "--keep-allele-order \\\n",
    "--make-bed \\\n",
    "--out neurod.LBD.controls.UNRELATED.chrX"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run QC for chr X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!cp -r ../Analysis.XWAS/scripts/ ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#! /bin/bash\n",
      "\n",
      "basePlinkFILENAME=$1\n",
      "email=$2\n",
      "\n",
      "# Warning message to indicate argument requirement for script to run\n",
      "if [ $# -eq 2 ]\n",
      "then\n",
      "    echo \"QC_preimputation_v3_keepRelated_wgs.sh running\"\n",
      "    echo \"This script should be executed in biowulf. If not, please terminate job.\"\n",
      "    echo \"This script uses plink2 that requires a processor which supports AVX2/Haswell instructions.\"\n",
      "    echo \"To be used for chrX QC on unrelated samples and EURO cohort i.e. will not perform ancestry check. Will assume samples are of one ancestry background.\"\n",
      "    echo \"Notification email will be sent when task is complete.\"\n",
      "else\n",
      "    echo \"Need plink input file for chr X merged case/control data and email\"\n",
      "    echo \"How to use: sh ../scripts/QC_preimputation_v3_UNRELATED_wgs_chrX.sh \\$basePlinkFILENAME \\$email\"\n",
      "    echo \"Note: This script is written to be executed in biowulf.\"\n",
      "    exit\n",
      "fi\n",
      "\n",
      "### How to use: sh QC_preimputation_v3_UNRELATED_wgs_chrX.sh $plinkINPUT $outputNAME $email\n",
      "### How to run as a swarm job. Be sure to specify processor that supports AVX2/Haswell\n",
      "# echo \"sh ../scripts/QC_preimputation_v3_UNRELATED_wgs_chrX.sh $plinkINPUT $email\" > qc.swarm\n",
      "# swarm --file qc.swarm --logdir swarmOE_qc --gres=lscratch:200 -g 120 -t auto --time 03:00:00 --sbatch '--constraint='\"'\"'x2680\\|x2695'\"'\"''\n",
      "\n",
      "###################\n",
      "### QC OVERVIEW ###\n",
      "###################\n",
      "\n",
      "### PART 1of2 ###\n",
      "\n",
      "# STEP 1  : SAMPLE-LEVEL FILTERING (HETEROZYGOSITY, CALL RATE, GENDER CHECK, ANCESTRY)\n",
      "# STEP 1.1: Sample-level filtering (Gender check)\n",
      "\n",
      "\n",
      "# STEP 2  : VARIANT-LEVEL FILTERING PER SEX (Variant inclusion criteria to account for genotyping batch differences; then make final bfiles with --geno 0.05)\n",
      "# STEP 2.1: Case/control nonrandom missingness test \\(i.e. exclude SNPs with P \\< 1E-4\\)\n",
      "# STEP 2.2: Haplotype-based test for non-random missing genotype data \\(i.e. exclude SNPs with P \\< 1E-4\\)\n",
      "# STEP 2.3: Run --hwe (females controls); filter to keep variants with midp > 0.0001\n",
      "# STEP 2.4: Remove variants that failed hwe from males and female cohort, --geno 0.02 to account for final variant missingness; and keep variants with maf > 0.01\n",
      "\n",
      "\n",
      "# STEP 3  : MERGE MALE & FEMALE QC-ED BINARIES; RUN ADDITIONAL QC\n",
      "# STEP 3.1: Merge female and male QCed binaries\n",
      "# STEP 3.2: Remove SNPs with significantly different missingness rates between male and female controls (--test-missing ; code male = 1/\"control\" and female = 2/\"case\")\n",
      "# STEP 3.3: Remove SNPs with significantly different MAF between male and female samples in the control group (--model-trend  ; code male = 1/\"control\" and female = 2/\"case\")\n",
      "# STEP 3.4: Remove SNPs in the pseudoautosomal regions (PARs)\n",
      "\n",
      "\n",
      "# Load necessary modules on biowulf\n",
      "module load plink/1.9.0-beta4.4\n",
      "module load R/3.5.2\n",
      "module load GCTA/1.92.0beta3\n",
      "module load flashpca/2.0\n",
      "\n",
      "# Create new directory for QC intermediate files\n",
      "mkdir males\n",
      "mkdir females\n",
      "mkdir males/QC\n",
      "mkdir females/QC\n",
      "\n",
      "\n",
      "# Date stamp to record when the script run was started\n",
      "start_date=$(date)\n",
      "echo \"Start date and time = $start_date\"\n",
      "res1=$(date +%s.%N)\n",
      "\n",
      "\n",
      "### Sample cleaning inclusion:\n",
      "\n",
      "# NOTE do all samples have a gender??\n",
      "# NOTE do all your samples have an affection status??\n",
      "\n",
      "# STEP 1: SAMPLE-LEVEL FILTERING (GENDER CHECK)\n",
      "\n",
      "\n",
      "echo\n",
      "echo STEP 1.1: Sample-level filtering \\(Gender check, remove samples that failed gender check\\)\n",
      "echo\n",
      "# PAR_hg38 =  --chr 23 --from-bp 2781479 --to-bp 155701383 --maf 0.05 --geno 0.05 --hwe 1E-5\n",
      "\n",
      "for sex in {males,females}\n",
      "do\n",
      "plink \\\n",
      "--bfile $basePlinkFILENAME \\\n",
      "--filter-${sex} \\\n",
      "--check-sex 0.25 0.75 --maf 0.05 \\\n",
      "--out ${sex}/QC/gender_check1;\n",
      "\n",
      "plink \\\n",
      "--bfile $basePlinkFILENAME \\\n",
      "--filter-${sex} \\\n",
      "--chr 23 \\\n",
      "--from-bp 2781479 \\\n",
      "--to-bp 155701383 \\\n",
      "--maf 0.05 \\\n",
      "--geno 0.05 \\\n",
      "--hwe 1E-5 \\\n",
      "--check-sex 0.25 0.75 \\\n",
      "--out ${sex}/QC/gender_check2\n",
      "done\n",
      "\n",
      "###  Make list of samples (FID, IID) that failed gender check\n",
      "for sex in {males,females}\n",
      "do\n",
      "grep \"PROBLEM\" ${sex}/QC/gender_check1.sexcheck | awk '{print $1\"\\t\"$2}' > ${sex}/QC/failedGenderCheck_samples_to_remove.txt\n",
      "grep \"PROBLEM\" ${sex}/QC/gender_check2.sexcheck | awk '{print $1\"\\t\"$2}' >> ${sex}/QC/failedGenderCheck_samples_to_remove.txt\n",
      "sort ${sex}/QC/failedGenderCheck_samples_to_remove.txt | uniq > ${sex}/failedGenderCheck_samples_to_remove_noDups.txt\n",
      "done\n",
      "\n",
      "### Remove samples --> remake plink2 binary files\n",
      "for sex in {males,females}\n",
      "do\n",
      "./scripts/plink2 \\\n",
      "--bfile $basePlinkFILENAME \\\n",
      "--filter-${sex} \\\n",
      "--remove ${sex}/failedGenderCheck_samples_to_remove_noDups.txt \\\n",
      "--make-bed \\\n",
      "--out ${sex}/QC/after_gender_chrX\n",
      "done\n",
      "\n",
      "\n",
      "\n",
      "# STEP 2: Variant inclusion criteria to account for genotyping batch differences; then make final bfiles with --geno 0.05\n",
      "echo\n",
      "echo STEP 2. Variant inclusion criteria to account for genotyping batch differences\n",
      "echo STEP 2.1: Case/control nonrandom missingness test \\(i.e. exclude SNPs with P \\< 1E-4\\)\n",
      "echo\n",
      "\n",
      "for sex in {males,females}\n",
      "do\n",
      "plink \\\n",
      "--bfile ${sex}/QC/after_gender_chrX \\\n",
      "--filter-${sex} \\\n",
      "--test-missing \\\n",
      "--allow-no-sex \\\n",
      "--out ${sex}/QC/missing_snps_chrX;\n",
      "\n",
      "awk '{if ($5 <= 0.0001) print $2}' ${sex}/QC/missing_snps_chrX.missing > ${sex}/QC/VARIANT_TEST_MISSING_SNPS_1E4_chrX.txt\n",
      "\n",
      "plink \\\n",
      "--bfile ${sex}/QC/after_gender_chrX \\\n",
      "--filter-${sex} \\\n",
      "--exclude ${sex}/QC/VARIANT_TEST_MISSING_SNPS_1E4_chrX.txt \\\n",
      "--allow-no-sex \\\n",
      "--make-bed \\\n",
      "--out ${sex}/QC/after_gender_CaseConMissingness_chrX;\n",
      "done\n",
      "\n",
      "\n",
      "echo\n",
      "echo STEP 2.2: Haplotype-based test for non-random missing genotype data \\(i.e. exclude SNPs with P \\< 1E-4\\)\n",
      "echo\n",
      "\n",
      "for sex in {males,females}\n",
      "do\n",
      "plink \\\n",
      "--bfile ${sex}/QC/after_gender_CaseConMissingness_chrX \\\n",
      "--filter-${sex} \\\n",
      "--test-mishap \\\n",
      "--allow-no-sex \\\n",
      "--out ${sex}/QC/missing_hap_chrX;\n",
      "\n",
      "awk '{if ($8 <= 0.0001) print $9}' ${sex}/QC/missing_hap_chrX.missing.hap > ${sex}/QC/HAPLOTYPE_TEST_MISSING_SNPS_1E4_chrX.txt;\n",
      "sed 's/|/\\n/g' ${sex}/QC/HAPLOTYPE_TEST_MISSING_SNPS_1E4_chrX.txt > ${sex}/QC/HAPLOTYPE_TEST_MISSING_SNPS_1E4_chrX_final.txt;\n",
      "\n",
      "plink \\\n",
      "--bfile ${sex}/QC/after_gender_CaseConMissingness_chrX \\\n",
      "--filter-${sex} \\\n",
      "--exclude ${sex}/QC/HAPLOTYPE_TEST_MISSING_SNPS_1E4_chrX_final.txt \\\n",
      "--allow-no-sex \\\n",
      "--make-bed \\\n",
      "--out ${sex}/QC/after_gender_CaseConMissingness_HaploMissingness_chrX;\n",
      "done\n",
      "\n",
      "\n",
      "echo\n",
      "echo \"STEP 2.3: Run --hwe (females controls); filter to keep variants with midp > 0.0001\"\n",
      "echo\n",
      "\n",
      "for sex in females\n",
      "do\n",
      "plink \\\n",
      "--bfile ${sex}/QC/after_gender_CaseConMissingness_HaploMissingness_chrX \\\n",
      "--filter-${sex} \\\n",
      "--filter-controls \\\n",
      "--hwe 0.0001 midp \\\n",
      "--write-snplist \\\n",
      "--out ${sex}/VarsPassHWE.${sex}_controls.1e-4\n",
      "done\n",
      "\n",
      "\n",
      "\n",
      "echo\n",
      "echo \"STEP 2.4: Remove variants that failed hwe from males and female cohort, --geno 0.02 to account for final variant missingness\"\n",
      "echo\n",
      "\n",
      "### NOTE: if prepping data for imputation; apply filter of --maf 0.005 so that the set of submitted snps is semi-common\n",
      "\n",
      "for sex in {males,females}\n",
      "do\n",
      "./scripts/plink2 \\\n",
      "--bfile ${sex}/QC/after_gender_CaseConMissingness_HaploMissingness_chrX \\\n",
      "--filter-${sex} \\\n",
      "--extract females/VarsPassHWE.females_controls.1e-4.snplist \\\n",
      "--geno 0.02 \\\n",
      "--make-bed \\\n",
      "--out ${sex}/FILTERED.${sex}.chrX\n",
      "done\n",
      "\n",
      "### Move all log files to a new folder\n",
      "for sex in {males,females}\n",
      "do\n",
      "mkdir ${sex}/QC/logFiles\n",
      "mv ${sex}/QC/*.log ${sex}/QC/logFiles\n",
      "done\n",
      "\n",
      "\n",
      "\n",
      "# STEP 3  : MERGE MALE & FEMALE QC-ED BINARIES; RUN ADDITIONAL QC\n",
      "echo\n",
      "echo STEP 3. MERGE MALE & FEMALE QC-ED BINARIES; RUN ADDITIONAL QC\n",
      "echo STEP 3.1: Merge female and male QCed binaries\n",
      "echo\n",
      "\n",
      "mkdir merged.clean\n",
      "\n",
      "## Make list of common variants shared between qc-ed female and male cohorts\n",
      "for sex in {males,females}\n",
      "do\n",
      "awk '{print $2}' ${sex}/FILTERED.${sex}.chrX.bim > ${sex}/postQC.vars.${sex}.txt\n",
      "done\n",
      "\n",
      "cat males/postQC.vars.males.txt females/postQC.vars.females.txt | sort | uniq -d > merged.clean/postQC.vars.common.txt\n",
      "\n",
      "## Subset variants from males and female cohorts\n",
      "for sex in {males,females}\n",
      "do\n",
      "plink \\\n",
      "--bfile ${sex}/FILTERED.${sex}.chrX \\\n",
      "--extract merged.clean/postQC.vars.common.txt \\\n",
      "--make-bed \\\n",
      "--out merged.clean/temp.${sex}.chrX\n",
      "done\n",
      "\n",
      "plink \\\n",
      "--bfile merged.clean/temp.males.chrX \\\n",
      "--bmerge merged.clean/temp.females.chrX \\\n",
      "--make-bed \\\n",
      "--out merged.clean/temp.males.females.chrX\n",
      "\n",
      "\n",
      "\n",
      "echo\n",
      "echo \"STEP 3.2: Remove SNPs with significantly different missingness rates between male and female controls (xwas --freqdiff-x 0.05)\"\n",
      "echo\n",
      "\n",
      "scripts/XWAS_v3.0/bin/xwas \\\n",
      "--bfile merged.clean/temp.males.females.chrX \\\n",
      "--xwas \\\n",
      "--freqdiff-x 0.05 \\\n",
      "--noweb \\\n",
      "--out merged.clean/freqdiff.chrX\n",
      "\n",
      "awk '{if ($9 < 0.0000001) print $2}' merged.clean/freqdiff.chrX.xtest > merged.clean/RatesDiffMissingness.bySex.controls.toRemove.txt\n",
      "\n",
      "\n",
      "plink \\\n",
      "--bfile merged.clean/temp.males.females.chrX \\\n",
      "--exclude merged.clean/RatesDiffMissingness.bySex.controls.toRemove.txt \\\n",
      "--make-bed \\\n",
      "--out merged.clean/temp.males.females.RatesDiffMissingness.chrX\n",
      "\n",
      "\n",
      "echo\n",
      "echo \"STEP 3.3: Remove SNPs with significantly different MAF between male and female samples in the control group (xwas --misdiff-x 0.05)\"\n",
      "echo\n",
      "\n",
      "scripts/XWAS_v3.0/bin/xwas \\\n",
      "--bfile merged.clean/temp.males.females.RatesDiffMissingness.chrX \\\n",
      "--xwas \\\n",
      "--missdiff-x 0.05 \\\n",
      "--noweb \\\n",
      "--out merged.clean/MAFDiffMissingness.chrX\n",
      "\n",
      "Rscript scripts/missdiff.R\n",
      "\n",
      "#!/usr/bin/env Rscript\n",
      "\n",
      "#require(data.table)\n",
      "#require(tidyverse)\n",
      "\n",
      "#data <- fread(\"merged.clean/MAFDiffMissingness.chrX.xmisstest\",header=T) %>% filter(Pvalue != is.na(Pvalue))\n",
      "#cutoff <- 0.05/dim(data)[1]\n",
      "#toRemove <- data %>% filter(Pvalue < cutoff) %>% select(SNP)\n",
      "#write.table(toRemove,\"merged.clean/MAFdiffMissingness.bySex.controls.toRemove.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=F)\n",
      "\n",
      "\n",
      "plink \\\n",
      "--bfile merged.clean/temp.males.females.RatesDiffMissingness.chrX \\\n",
      "--exclude merged.clean/MAFdiffMissingness.bySex.controls.toRemove.txt \\\n",
      "--make-bed \\\n",
      "--out merged.clean/temp.males.females.RatesDiffMissingness.MAFdiffMissingness.chrX\n",
      "\n",
      "\n",
      "\n",
      "echo\n",
      "echo \"STEP 3.4: Remove SNPs in the pseudoautosomal regions (PARs)\"\n",
      "echo\n",
      "\n",
      "plink \\\n",
      "--bfile merged.clean/temp.males.females.RatesDiffMissingness.MAFdiffMissingness.chrX \\\n",
      "--exclude range scripts/PAR_hg38_region.txt \\\n",
      "--make-bed \\\n",
      "--out merged.clean/FINAL.QC.males.females.noPAR.chrX\n",
      "\n",
      "\n",
      "### Move all log files to a new folder\n",
      "mkdir merged.clean/logFiles\n",
      "mv merged.clean/*.log merged.clean/logFiles\n",
      "\n",
      "\n",
      "### Remove intermediate files\n",
      "for sex in {males,females}\n",
      "do\n",
      "rm ${sex}/QC/after_gender_chrX.*\n",
      "rm ${sex}/QC/after_gender_CaseConMissingness_chrX.*\n",
      "rm ${sex}/QC/after_gender_CaseConMissingness_HaploMissingness_chrX.*\n",
      "done\n",
      "rm merged.clean/temp*\n",
      "\n",
      "\n",
      "\n",
      "# Gives runtime\n",
      "res2=$(date +%s.%N)\n",
      "dt=$(echo \"$res2 - $res1\" | bc)\n",
      "dd=$(echo \"$dt/86400\" | bc)\n",
      "dt2=$(echo \"$dt-86400*$dd\" | bc)\n",
      "dh=$(echo \"$dt2/3600\" | bc)\n",
      "dt3=$(echo \"$dt2-3600*$dh\" | bc)\n",
      "dm=$(echo \"$dt3/60\" | bc)\n",
      "ds=$(echo \"$dt3-60*$dm\" | bc)\n",
      "\n",
      "end_date=$(date)\n",
      "echo \"End date and time = $end_date\"\n",
      "printf \"Total runtime: %d:%02d:%02d:%02.4f\\n\" $dd $dh $dm $ds\n",
      "printf \"Total runtime: %d:%02d:%02d:%02.4f\\n\" $dd $dh $dm $ds > Runtime_summary.txt\n",
      "\n",
      "mailx -s \"biowulf swarm alert message: QC_preimputation_v3_UNRELATED_wgs_chrX TASK COMPLETE\" $email < Runtime_summary.txt\n",
      "sleep 10s\n",
      "rm Runtime_summary.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!cat ./scripts/QC_preimputation_v3_UNRELATED_wgs_chrX.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "sh ./scripts/QC_preimputation_v3_UNRELATED_wgs_chrX.sh neurod.LBD.controls.UNRELATED.chrX chiarp@mail.nih.gov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!cat /data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/scripts/QC_preimputation_v3_UNRELATED_wgs_chrX.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ***NEW 06-02-2023:*** Run step() for sex-stratified cohorts using genetic PCs generated from autosome chromosomes\n",
    "\n",
    "Note that the genetic PCs were generated and recorded in another notebook; refer to: `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/2024-01-10_Readme_ChromX_haplotypeblocks_estimation_PCA_calculations.ipynb`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# read in file - male/female samples\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "data <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE))\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED_clean.updated_covs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "dim(xwas)\n",
    "\n",
    "table(data_0$AFFECTION_STATUS,data_0$SEX)\n",
    "\n",
    "table(data$AFFECTION_STATUS,data$SEX)\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "# subset to females only\n",
    "females <- data %>% filter(SEX == 2)\n",
    "write.table(females,\"COVARIATES.freeze9.LBD.controls.UNRELATED.females_clean.updated_covs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(females$CASE,females$GENDER)\n",
    "\n",
    "# subset to males only\n",
    "males <- data %>% filter(SEX == 1)\n",
    "write.table(males,\"COVARIATES.freeze9.LBD.controls.UNRELATED.males_clean.updated_covs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(males$CASE,males$GENDER)\n",
    "\n",
    "# subset to female and male CASES only\n",
    "cases <- data %>% filter(AFFECTION_STATUS == 2)\n",
    "write.table(cases,\"COVARIATES.freeze9.LBD.controls.UNRELATED.cases_clean.updated_covs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(cases$CASE,cases$GENDER)\n",
    "\n",
    "# Run step() to determine which covariate is most important\n",
    "# model using glm() and then use step() to determine the model that has the lowest AIC\n",
    "# this tell us which covariates included in the model are important to use/for adjustment in analysis\n",
    "\n",
    "## linear age ##\n",
    "# for females only\n",
    "model_females <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = females, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model_females)\n",
    "step(model_females)\n",
    "\n",
    "# for males only\n",
    "model_males <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = males, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model_males)\n",
    "step(model_males)\n",
    "\n",
    "# for all male/female samples\n",
    "model <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model)\n",
    "step(model)\n",
    "\n",
    "# for all cases samples\n",
    "model <- glm(GENDER ~ CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = cases, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model)\n",
    "step(model)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ***NEW 12-08-2023:*** Run step() for sex-stratified cohorts using genetic PCs generated from X-chromosome\n",
    "\n",
    "Note that the genetic PCs were generated and recorded in another notebook; refer to: `http://localhost:38407/lab/workspaces/auto-n/tree/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/2024-01-10_Readme_ChromX_haplotypeblocks_estimation_PCA_calculations.ipynb`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# read in file - male/female samples\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "\n",
    "# remove autosomal PCs and replace with x-chromosome PCs\n",
    "temp <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE)) %>%\n",
    "        select(-PC1,-PC2,-PC3,-PC4,-PC5,-PC6,-PC7,-PC8,-PC9,-PC10)\n",
    "pcs_x <- fread(\"X-PCA_redo/both/pcs_pruned_chrX_forPCA\")\n",
    "data <- merge(temp, pcs_x, by=c(\"FID\",\"IID\"))\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED_clean.updated_covs_using_x-PCs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "dim(xwas)\n",
    "\n",
    "table(data_0$AFFECTION_STATUS,data_0$SEX)\n",
    "table(data$AFFECTION_STATUS,data$SEX)\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "model <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model)\n",
    "step(model)\n",
    "\n",
    "\n",
    "# read in file - females only\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "\n",
    "# remove autosomal PCs and replace with x-chromosome PCs\n",
    "temp <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE)) %>%\n",
    "        select(-PC1,-PC2,-PC3,-PC4,-PC5,-PC6,-PC7,-PC8,-PC9,-PC10)\n",
    "pcs_x <- fread(\"X-PCA_redo/females_only/pcs_pruned_chrX_forPCA\")\n",
    "data <- merge(temp, pcs_x, by=c(\"FID\",\"IID\"))\n",
    "head(data)\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED.females_clean.updated_covs_using_x-PCs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "model_females <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model_females)\n",
    "step(model_females)\n",
    "\n",
    "\n",
    "# read in file - males only\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "\n",
    "# remove autosomal PCs and replace with x-chromosome PCs\n",
    "temp <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE)) %>%\n",
    "        select(-PC1,-PC2,-PC3,-PC4,-PC5,-PC6,-PC7,-PC8,-PC9,-PC10)\n",
    "pcs_x <- fread(\"X-PCA_redo/males_only/pcs_pruned_chrX_forPCA\")\n",
    "data <- merge(temp, pcs_x, by=c(\"FID\",\"IID\"))\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED.males_clean.updated_covs_using_x-PCs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "model_males <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model_males)\n",
    "step(model_males)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ***NEW 12-19-2023:*** Run step() for sex-stratified cohorts using genetic PCs generated from merged autosomal and X-chromosome\n",
    "\n",
    "Note that the genetic PCs were generated and recorded in another notebook; refer to: `http://localhost:38407/lab/workspaces/auto-n/tree/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/2024-01-10_Readme_ChromX_haplotypeblocks_estimation_PCA_calculations.ipynb`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# read in file - male/female samples\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "\n",
    "# remove autosomal PCs and replace with x-chromosome PCs\n",
    "temp <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE)) %>%\n",
    "        select(-PC1,-PC2,-PC3,-PC4,-PC5,-PC6,-PC7,-PC8,-PC9,-PC10)\n",
    "pcs_x <- fread(\"X_autosomal-PCA/both/pcs_FILTERED_ALLchr_X_forPCA\")\n",
    "data <- merge(temp, pcs_x, by=c(\"FID\",\"IID\"))\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED_clean.updated_covs_using_x-autosomal-PCs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "dim(xwas)\n",
    "\n",
    "table(data_0$AFFECTION_STATUS,data_0$SEX)\n",
    "table(data$AFFECTION_STATUS,data$SEX)\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "model <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model)\n",
    "step(model)\n",
    "\n",
    "\n",
    "# read in file - females only\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "\n",
    "# remove autosomal PCs and replace with x-chromosome PCs\n",
    "temp <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE)) %>%\n",
    "        select(-PC1,-PC2,-PC3,-PC4,-PC5,-PC6,-PC7,-PC8,-PC9,-PC10)\n",
    "pcs_x <- fread(\"X_autosomal-PCA/females_only/pcs_FILTERED_ALLchr_X_forPCA\")\n",
    "data <- merge(temp, pcs_x, by=c(\"FID\",\"IID\"))\n",
    "head(data)\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED.females_clean.updated_covs_using_x-autosomal-PCs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "model_females <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model_females)\n",
    "step(model_females)\n",
    "\n",
    "\n",
    "# read in file - males only\n",
    "data_0 <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.GLM/GLM.LBD.4Kcontrols/COVARIATES.freeze9.LBD.controls.UNRELATED.txt\",header=T)\n",
    "xwas <- fread(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS/COVARIATES.freeze9.LBD.controls.UNRELATED.exclGENDER.forXWAS.txt\",header=T)\n",
    "\n",
    "# remove autosomal PCs and replace with x-chromosome PCs\n",
    "temp <- subset(data_0, data_0$IID %in% xwas$IID) %>% filter(!is.na(CONSENSUS_AGE)) %>%\n",
    "        select(-PC1,-PC2,-PC3,-PC4,-PC5,-PC6,-PC7,-PC8,-PC9,-PC10)\n",
    "pcs_x <- fread(\"X_autosomal-PCA/males_only/pcs_FILTERED_ALLchr_X_forPCA\")\n",
    "data <- merge(temp, pcs_x, by=c(\"FID\",\"IID\"))\n",
    "\n",
    "# test out squred age for covariate because the lambdas a quite inflated\n",
    "## ref: https://stats.stackexchange.com/questions/19823/why-would-one-use-age-squared-as-a-covariate-in-a-genetic-association-study\n",
    "data$AGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$AGExAGExAGE <- data$CONSENSUS_AGE * data$CONSENSUS_AGE * data$CONSENSUS_AGE\n",
    "data$log_AGE <- log(data$CONSENSUS_AGE)\n",
    "data$SEXxAGE <- data$GENDER * data$CONSENSUS_AGE\n",
    "data$CASE <- data$AFFECTION_STATUS - 1\n",
    "write.table(data,\"COVARIATES.freeze9.LBD.controls.UNRELATED.males_clean.updated_covs_using_x-autosomal-PCs.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "table(data$CASE,data$GENDER)\n",
    "\n",
    "model_males <- glm(CASE ~ GENDER + CONSENSUS_AGE + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 , data = data, family = \"binomial\"(link = \"logit\"))\n",
    "summary(model_males)\n",
    "step(model_males)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.10",
   "language": "python",
   "name": "py3.10"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "388.508px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
