{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# XWAS/COLOC using GTEX v8 brain dataset\n",
    "\n",
    "**Start date:** 2023-12-28\n",
    "\n",
    "**Updated date:** 2023-12-28\n",
    "\n",
    "**Author(s):** Ruth Chia\n",
    "\n",
    "**Working directory on biowulf:** `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/vf/users/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ***UPDATED 12-28-2023:*** COLOC with GTEX v8 brain dataset\n",
    "\n",
    "Some notes:\n",
    "1. Downloaded 13 brain eQTL associations from GTEX v8. Location of formatted eQTL summ stats for 13 brain regions are: `/data/NDRS_LNG/GTEX_v8/GTEX_v8_chrX_brain/eQTL/Brain_*.v8.EUR.allpairs.chrX.txt`\n",
    "\n",
    "\n",
    "2. The 13 brain regions are:\n",
    "- `Brain_Amygdala`\n",
    "- `Brain_Anterior_cingulate_cortex_BA24`\n",
    "- `Brain_Caudate_basal_ganglia`\n",
    "- `Brain_Cerebellar_Hemisphere`\n",
    "- `Brain_Cerebellum`\n",
    "- `Brain_Cortex`\n",
    "- `Brain_Frontal_Cortex_BA9`\n",
    "- `Brain_Hippocampus`\n",
    "- `Brain_Hypothalamus`\n",
    "- `Brain_Nucleus_accumbens_basal_ganglia`\n",
    "- `Brain_Putamen_basal_ganglia`\n",
    "- `Brain_Spinal_cord_cervical_c-1`\n",
    "- `Brain_Substantia_nigra`\n",
    "\n",
    "    \n",
    "3. XWAS summary stats to run coloc on are (only for regions of interest with significant/subsignificant (significance threshold was set at 7.94x10-6) hit +/- 0.5Mb i.e. according the the coloc recommendation on `https://chr1swallace.github.io/coloc/articles/a02_data.html`):\n",
    "    - females (not conditioned on ApoE): `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/females_only_x-autosomal-pc/toMeta.LBD.controls.UNRELATED.females.maf0.01overall.hg38.chrX.rsid_REFORMATTED.txt`\n",
    "    - females (conditioned on ApoE): `/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/cond_ApoE4_x-autosomal-pc/females_only/toMeta.LBD.controls.UNRELATED.females.maf0.01overall.hg38.chrX.rsid_REFORMATTED.txt`\n",
    "    \n",
    "____\n",
    "\n",
    "**What I would need to do:**\n",
    "1. Filter summ stats to include significant/subsignificant hit from summ stats - then flank out by 1Mb from top index variant.\n",
    "2. Harmonize eQTL summ stats to include the same set of variants and to make sure that the variant ID is the same (can keep it to chr:pos:ref:alt' format.\n",
    "3. Prep not summ stats for XWAS and eQTL so that it contains the necessary columns to run coloc using the coloc R package (tutorial to run coloc on two summ stats can be found: `https://chr1swallace.github.io/coloc/articles/a02_data.html`)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!mkdir Analysis.COLOC_redo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!mkdir Analysis.COLOC_redo/females_only\n",
    "!mkdir Analysis.COLOC_redo/females_only_cond_ApoE4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!mkdir Analysis.COLOC_redo/GTEXv8_eQTL_brains\n",
    "!mkdir Analysis.COLOC_redo/GTEXv8_sQTL_brains"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create subset of snp region to run coloc on \n",
    "\n",
    "#### females-only xwas (unconditioned on APOE4)\n",
    "\n",
    "top index variants from Table 1 of paper are:\n",
    "- rs141773145 (pos = 19,513,849)\n",
    "- rs12860838 (pos = 117,373,825)\n",
    "\n",
    "use these variants and flank out by 1Mb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['CHROM', 'POS', 'ID', 'rsID', 'EffectAllele', 'OtherAllele', 'BETA',\n",
      "       'BETA_SE', 'OR', 'OR_low95', 'OR_high95', 'P', 'MAF_EffectAllele',\n",
      "       'MAF_EffectAllele_CASE', 'MAF_EffectAllele_CTRL', 'CT_EffectAllele',\n",
      "       'CT_EffectAllele_CASE', 'CT_EffectAllele_CTRL', 'ALLELE_CT',\n",
      "       'CASE_ALLELE_CT', 'CTRL_ALLELE_CT'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "females_uncond = pd.read_csv(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/females_only_x-autosomal-pc/toMeta.LBD.controls.UNRELATED.females.maf0.01overall.hg38.chrX.rsid_REFORMATTED.txt\",sep=\"\\t\")\n",
    "print(females_uncond.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>POS</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>7.933355e+07</td>\n",
       "      <td>4.563753e+07</td>\n",
       "      <td>2.781514e+06</td>\n",
       "      <td>3.614308e+07</td>\n",
       "      <td>8.358717e+07</td>\n",
       "      <td>1.193624e+08</td>\n",
       "      <td>1.557006e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BETA</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>-6.831485e-03</td>\n",
       "      <td>1.435296e-01</td>\n",
       "      <td>-1.369221e+00</td>\n",
       "      <td>-6.981106e-02</td>\n",
       "      <td>-2.739750e-03</td>\n",
       "      <td>6.471038e-02</td>\n",
       "      <td>9.655455e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BETA_SE</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.180516e-01</td>\n",
       "      <td>6.720031e-02</td>\n",
       "      <td>5.539100e-02</td>\n",
       "      <td>6.369070e-02</td>\n",
       "      <td>8.736010e-02</td>\n",
       "      <td>1.600105e-01</td>\n",
       "      <td>4.408130e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.003327e+00</td>\n",
       "      <td>1.435414e-01</td>\n",
       "      <td>2.543050e-01</td>\n",
       "      <td>9.325705e-01</td>\n",
       "      <td>9.972740e-01</td>\n",
       "      <td>1.066850e+00</td>\n",
       "      <td>2.626220e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR_low95</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>8.030415e-01</td>\n",
       "      <td>1.415592e-01</td>\n",
       "      <td>1.081022e-01</td>\n",
       "      <td>7.350166e-01</td>\n",
       "      <td>8.360580e-01</td>\n",
       "      <td>8.983707e-01</td>\n",
       "      <td>1.628731e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR_high95</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.275058e+00</td>\n",
       "      <td>2.717356e-01</td>\n",
       "      <td>5.841018e-01</td>\n",
       "      <td>1.111900e+00</td>\n",
       "      <td>1.193951e+00</td>\n",
       "      <td>1.344726e+00</td>\n",
       "      <td>4.234603e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>4.850853e-01</td>\n",
       "      <td>2.933628e-01</td>\n",
       "      <td>3.436790e-05</td>\n",
       "      <td>2.249820e-01</td>\n",
       "      <td>4.805790e-01</td>\n",
       "      <td>7.400285e-01</td>\n",
       "      <td>9.999950e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF_EffectAllele</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.689946e-01</td>\n",
       "      <td>1.479782e-01</td>\n",
       "      <td>1.013960e-02</td>\n",
       "      <td>3.407580e-02</td>\n",
       "      <td>1.228390e-01</td>\n",
       "      <td>2.847410e-01</td>\n",
       "      <td>5.000000e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF_EffectAllele_CASE</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.685472e-01</td>\n",
       "      <td>1.480764e-01</td>\n",
       "      <td>3.164560e-03</td>\n",
       "      <td>3.375530e-02</td>\n",
       "      <td>1.228900e-01</td>\n",
       "      <td>2.850740e-01</td>\n",
       "      <td>5.290080e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF_EffectAllele_CTRL</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.692005e-01</td>\n",
       "      <td>1.480586e-01</td>\n",
       "      <td>7.038830e-03</td>\n",
       "      <td>3.398060e-02</td>\n",
       "      <td>1.223300e-01</td>\n",
       "      <td>2.839810e-01</td>\n",
       "      <td>5.128640e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CT_EffectAllele</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.016672e+03</td>\n",
       "      <td>8.902371e+02</td>\n",
       "      <td>6.100000e+01</td>\n",
       "      <td>2.050000e+02</td>\n",
       "      <td>7.390000e+02</td>\n",
       "      <td>1.713000e+03</td>\n",
       "      <td>3.008000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CT_EffectAllele_CASE</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>3.195655e+02</td>\n",
       "      <td>2.807528e+02</td>\n",
       "      <td>6.000000e+00</td>\n",
       "      <td>6.400000e+01</td>\n",
       "      <td>2.330000e+02</td>\n",
       "      <td>5.405000e+02</td>\n",
       "      <td>1.003000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CT_EffectAllele_CTRL</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>6.971060e+02</td>\n",
       "      <td>6.100016e+02</td>\n",
       "      <td>2.900000e+01</td>\n",
       "      <td>1.400000e+02</td>\n",
       "      <td>5.040000e+02</td>\n",
       "      <td>1.170000e+03</td>\n",
       "      <td>2.113000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>6.016000e+03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.016000e+03</td>\n",
       "      <td>6.016000e+03</td>\n",
       "      <td>6.016000e+03</td>\n",
       "      <td>6.016000e+03</td>\n",
       "      <td>6.016000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASE_ALLELE_CT</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>1.896000e+03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.896000e+03</td>\n",
       "      <td>1.896000e+03</td>\n",
       "      <td>1.896000e+03</td>\n",
       "      <td>1.896000e+03</td>\n",
       "      <td>1.896000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTRL_ALLELE_CT</th>\n",
       "      <td>258247.0</td>\n",
       "      <td>4.120000e+03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.120000e+03</td>\n",
       "      <td>4.120000e+03</td>\n",
       "      <td>4.120000e+03</td>\n",
       "      <td>4.120000e+03</td>\n",
       "      <td>4.120000e+03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          count          mean           std           min  \\\n",
       "POS                    258247.0  7.933355e+07  4.563753e+07  2.781514e+06   \n",
       "BETA                   258247.0 -6.831485e-03  1.435296e-01 -1.369221e+00   \n",
       "BETA_SE                258247.0  1.180516e-01  6.720031e-02  5.539100e-02   \n",
       "OR                     258247.0  1.003327e+00  1.435414e-01  2.543050e-01   \n",
       "OR_low95               258247.0  8.030415e-01  1.415592e-01  1.081022e-01   \n",
       "OR_high95              258247.0  1.275058e+00  2.717356e-01  5.841018e-01   \n",
       "P                      258247.0  4.850853e-01  2.933628e-01  3.436790e-05   \n",
       "MAF_EffectAllele       258247.0  1.689946e-01  1.479782e-01  1.013960e-02   \n",
       "MAF_EffectAllele_CASE  258247.0  1.685472e-01  1.480764e-01  3.164560e-03   \n",
       "MAF_EffectAllele_CTRL  258247.0  1.692005e-01  1.480586e-01  7.038830e-03   \n",
       "CT_EffectAllele        258247.0  1.016672e+03  8.902371e+02  6.100000e+01   \n",
       "CT_EffectAllele_CASE   258247.0  3.195655e+02  2.807528e+02  6.000000e+00   \n",
       "CT_EffectAllele_CTRL   258247.0  6.971060e+02  6.100016e+02  2.900000e+01   \n",
       "ALLELE_CT              258247.0  6.016000e+03  0.000000e+00  6.016000e+03   \n",
       "CASE_ALLELE_CT         258247.0  1.896000e+03  0.000000e+00  1.896000e+03   \n",
       "CTRL_ALLELE_CT         258247.0  4.120000e+03  0.000000e+00  4.120000e+03   \n",
       "\n",
       "                                25%           50%           75%           max  \n",
       "POS                    3.614308e+07  8.358717e+07  1.193624e+08  1.557006e+08  \n",
       "BETA                  -6.981106e-02 -2.739750e-03  6.471038e-02  9.655455e-01  \n",
       "BETA_SE                6.369070e-02  8.736010e-02  1.600105e-01  4.408130e-01  \n",
       "OR                     9.325705e-01  9.972740e-01  1.066850e+00  2.626220e+00  \n",
       "OR_low95               7.350166e-01  8.360580e-01  8.983707e-01  1.628731e+00  \n",
       "OR_high95              1.111900e+00  1.193951e+00  1.344726e+00  4.234603e+00  \n",
       "P                      2.249820e-01  4.805790e-01  7.400285e-01  9.999950e-01  \n",
       "MAF_EffectAllele       3.407580e-02  1.228390e-01  2.847410e-01  5.000000e-01  \n",
       "MAF_EffectAllele_CASE  3.375530e-02  1.228900e-01  2.850740e-01  5.290080e-01  \n",
       "MAF_EffectAllele_CTRL  3.398060e-02  1.223300e-01  2.839810e-01  5.128640e-01  \n",
       "CT_EffectAllele        2.050000e+02  7.390000e+02  1.713000e+03  3.008000e+03  \n",
       "CT_EffectAllele_CASE   6.400000e+01  2.330000e+02  5.405000e+02  1.003000e+03  \n",
       "CT_EffectAllele_CTRL   1.400000e+02  5.040000e+02  1.170000e+03  2.113000e+03  \n",
       "ALLELE_CT              6.016000e+03  6.016000e+03  6.016000e+03  6.016000e+03  \n",
       "CASE_ALLELE_CT         1.896000e+03  1.896000e+03  1.896000e+03  1.896000e+03  \n",
       "CTRL_ALLELE_CT         4.120000e+03  4.120000e+03  4.120000e+03  4.120000e+03  "
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "females_uncond.describe().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>rsID</th>\n",
       "      <th>EffectAllele</th>\n",
       "      <th>OtherAllele</th>\n",
       "      <th>BETA</th>\n",
       "      <th>BETA_SE</th>\n",
       "      <th>OR</th>\n",
       "      <th>OR_low95</th>\n",
       "      <th>...</th>\n",
       "      <th>P</th>\n",
       "      <th>MAF_EffectAllele</th>\n",
       "      <th>MAF_EffectAllele_CASE</th>\n",
       "      <th>MAF_EffectAllele_CTRL</th>\n",
       "      <th>CT_EffectAllele</th>\n",
       "      <th>CT_EffectAllele_CASE</th>\n",
       "      <th>CT_EffectAllele_CTRL</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>CASE_ALLELE_CT</th>\n",
       "      <th>CTRL_ALLELE_CT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>X</td>\n",
       "      <td>2781514</td>\n",
       "      <td>chrX:2781514:C:A</td>\n",
       "      <td>rs311165</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.025560</td>\n",
       "      <td>0.058254</td>\n",
       "      <td>0.974764</td>\n",
       "      <td>0.869586</td>\n",
       "      <td>...</td>\n",
       "      <td>0.660837</td>\n",
       "      <td>0.416057</td>\n",
       "      <td>0.410338</td>\n",
       "      <td>0.418689</td>\n",
       "      <td>2503</td>\n",
       "      <td>778</td>\n",
       "      <td>1725</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X</td>\n",
       "      <td>2781604</td>\n",
       "      <td>chrX:2781604:G:T</td>\n",
       "      <td>rs28579419</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.022065</td>\n",
       "      <td>0.082083</td>\n",
       "      <td>1.022310</td>\n",
       "      <td>0.870387</td>\n",
       "      <td>...</td>\n",
       "      <td>0.788093</td>\n",
       "      <td>0.138630</td>\n",
       "      <td>0.143460</td>\n",
       "      <td>0.136408</td>\n",
       "      <td>834</td>\n",
       "      <td>272</td>\n",
       "      <td>562</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS                ID        rsID EffectAllele OtherAllele  \\\n",
       "0     X  2781514  chrX:2781514:C:A    rs311165            A           C   \n",
       "1     X  2781604  chrX:2781604:G:T  rs28579419            T           G   \n",
       "\n",
       "       BETA   BETA_SE        OR  OR_low95  ...         P  MAF_EffectAllele  \\\n",
       "0 -0.025560  0.058254  0.974764  0.869586  ...  0.660837          0.416057   \n",
       "1  0.022065  0.082083  1.022310  0.870387  ...  0.788093          0.138630   \n",
       "\n",
       "   MAF_EffectAllele_CASE  MAF_EffectAllele_CTRL  CT_EffectAllele  \\\n",
       "0               0.410338               0.418689             2503   \n",
       "1               0.143460               0.136408              834   \n",
       "\n",
       "   CT_EffectAllele_CASE  CT_EffectAllele_CTRL  ALLELE_CT  CASE_ALLELE_CT  \\\n",
       "0                   778                  1725       6016            1896   \n",
       "1                   272                   562       6016            1896   \n",
       "\n",
       "   CTRL_ALLELE_CT  \n",
       "0            4120  \n",
       "1            4120  \n",
       "\n",
       "[2 rows x 21 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "females_uncond.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "females_uncond_indexvar_1_flank1Mb = females_uncond[(females_uncond.POS > 19513849 - 1000000) & (females_uncond.POS < 19513849 + 1000000)]\n",
    "females_uncond_indexvar_2_flank1Mb = females_uncond[(females_uncond.POS > 117373825 - 1000000) & (females_uncond.POS < 117373825 + 1000000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "females_uncond_indexvar_1_flank1Mb.to_csv(\"Analysis.COLOC_redo/females_only/XWAS_summ_stats_rs141773145_flank1Mb_no-maf-filter.txt\",sep=\"\\t\",index=False,header=True)\n",
    "females_uncond_indexvar_2_flank1Mb.to_csv(\"Analysis.COLOC_redo/females_only/XWAS_summ_stats_rs12860838_flank1Mb_no-maf-filter.txt\",sep=\"\\t\",index=False,header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2419, 21)\n",
      "(5314, 21)\n"
     ]
    }
   ],
   "source": [
    "print(females_uncond_indexvar_1_flank1Mb.shape)\n",
    "print(females_uncond_indexvar_2_flank1Mb.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### females-only xwas (conditioned on APOE4)\n",
    "\n",
    "top index variants from Table 1 of paper are:\n",
    "- rs141773145(pos = 19,513,849)\n",
    "- rs6648060 (pos = 76,575,769)\n",
    "- rs141193614 (pos = 98,584,345)\n",
    "- rs12860838 (pos = 117,373,825)\n",
    "\n",
    "use these variants and flank out by 0.5Mb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['CHROM', 'POS', 'ID', 'rsID', 'EffectAllele', 'OtherAllele', 'BETA',\n",
      "       'BETA_SE', 'OR', 'OR_low95', 'OR_high95', 'P', 'MAF_EffectAllele',\n",
      "       'MAF_EffectAllele_CASE', 'MAF_EffectAllele_CTRL', 'CT_EffectAllele',\n",
      "       'CT_EffectAllele_CASE', 'CT_EffectAllele_CTRL', 'ALLELE_CT',\n",
      "       'CASE_ALLELE_CT', 'CTRL_ALLELE_CT'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "females_cond = pd.read_csv(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/cond_ApoE4_x-autosomal-pc/females_only/toMeta.LBD.controls.UNRELATED.females.maf0.01overall.hg38.chrX.rsid_REFORMATTED.txt\",sep=\"\\t\")\n",
    "print(females_cond.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "females_cond_indexvar_1_flank1Mb = females_cond[(females_cond.POS > 19513849 - 1000000) & (females_cond.POS < 19513849 + 1000000)]\n",
    "females_cond_indexvar_2_flank1Mb = females_cond[(females_cond.POS > 76575769 - 1000000) & (females_cond.POS < 76575769 + 1000000)]\n",
    "females_cond_indexvar_3_flank1Mb = females_cond[(females_cond.POS > 98584345 - 1000000) & (females_cond.POS < 98584345 + 1000000)]\n",
    "females_cond_indexvar_4_flank1Mb = females_cond[(females_cond.POS > 117373825 - 1000000) & (females_cond.POS < 117373825 + 1000000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "females_cond_indexvar_1_flank1Mb.to_csv(\"Analysis.COLOC_redo/females_only_cond_ApoE4/XWAS_summ_stats_rs141773145_flank1Mb_no-maf-filter.txt\",sep=\"\\t\",index=False,header=True)\n",
    "females_cond_indexvar_2_flank1Mb.to_csv(\"Analysis.COLOC_redo/females_only_cond_ApoE4/XWAS_summ_stats_rs6648060_flank1Mb_no-maf-filter.txt\",sep=\"\\t\",index=False,header=True)\n",
    "females_cond_indexvar_3_flank1Mb.to_csv(\"Analysis.COLOC_redo/females_only_cond_ApoE4/XWAS_summ_stats_rs141193614_flank1Mb_no-maf-filter.txt\",sep=\"\\t\",index=False,header=True)\n",
    "females_cond_indexvar_4_flank1Mb.to_csv(\"Analysis.COLOC_redo/females_only_cond_ApoE4/XWAS_summ_stats_rs12860838_flank1Mb_no-maf-filter.txt\",sep=\"\\t\",index=False,header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2422, 21)\n",
      "(5192, 21)\n",
      "(2416, 21)\n",
      "(5315, 21)\n"
     ]
    }
   ],
   "source": [
    "print(females_cond_indexvar_1_flank1Mb.shape)\n",
    "print(females_cond_indexvar_2_flank1Mb.shape)\n",
    "print(females_cond_indexvar_3_flank1Mb.shape)\n",
    "print(females_cond_indexvar_4_flank1Mb.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Harmonize eQTL dataset\n",
    "\n",
    "Use the unconditioned XWAS from females only to reformat variant ID.\n",
    "\n",
    "The conditioned XWAS summ stats should have the same REF/ALT designation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(258247, 21)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "xwas = pd.read_csv(\"/data/ALS_50k/DementiaSeq.TopmedJointCalled.June2020/LBD/Analysis.XWAS_GLM/females_only_x-autosomal-pc/toMeta.LBD.controls.UNRELATED.females.maf0.01overall.hg38.chrX.rsid_REFORMATTED.txt\",sep=\"\\t\")     \n",
    "print(xwas.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>rsID</th>\n",
       "      <th>EffectAllele</th>\n",
       "      <th>OtherAllele</th>\n",
       "      <th>BETA</th>\n",
       "      <th>BETA_SE</th>\n",
       "      <th>OR</th>\n",
       "      <th>OR_low95</th>\n",
       "      <th>...</th>\n",
       "      <th>P</th>\n",
       "      <th>MAF_EffectAllele</th>\n",
       "      <th>MAF_EffectAllele_CASE</th>\n",
       "      <th>MAF_EffectAllele_CTRL</th>\n",
       "      <th>CT_EffectAllele</th>\n",
       "      <th>CT_EffectAllele_CASE</th>\n",
       "      <th>CT_EffectAllele_CTRL</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>CASE_ALLELE_CT</th>\n",
       "      <th>CTRL_ALLELE_CT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>X</td>\n",
       "      <td>2781514</td>\n",
       "      <td>chrX:2781514:C:A</td>\n",
       "      <td>rs311165</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.025560</td>\n",
       "      <td>0.058254</td>\n",
       "      <td>0.974764</td>\n",
       "      <td>0.869586</td>\n",
       "      <td>...</td>\n",
       "      <td>0.660837</td>\n",
       "      <td>0.416057</td>\n",
       "      <td>0.410338</td>\n",
       "      <td>0.418689</td>\n",
       "      <td>2503</td>\n",
       "      <td>778</td>\n",
       "      <td>1725</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X</td>\n",
       "      <td>2781604</td>\n",
       "      <td>chrX:2781604:G:T</td>\n",
       "      <td>rs28579419</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.022065</td>\n",
       "      <td>0.082083</td>\n",
       "      <td>1.022310</td>\n",
       "      <td>0.870387</td>\n",
       "      <td>...</td>\n",
       "      <td>0.788093</td>\n",
       "      <td>0.138630</td>\n",
       "      <td>0.143460</td>\n",
       "      <td>0.136408</td>\n",
       "      <td>834</td>\n",
       "      <td>272</td>\n",
       "      <td>562</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>X</td>\n",
       "      <td>2781635</td>\n",
       "      <td>chrX:2781635:G:A</td>\n",
       "      <td>rs60075487</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.114979</td>\n",
       "      <td>0.077130</td>\n",
       "      <td>1.121850</td>\n",
       "      <td>0.964451</td>\n",
       "      <td>...</td>\n",
       "      <td>0.136048</td>\n",
       "      <td>0.153757</td>\n",
       "      <td>0.165084</td>\n",
       "      <td>0.148544</td>\n",
       "      <td>925</td>\n",
       "      <td>313</td>\n",
       "      <td>612</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>X</td>\n",
       "      <td>2781927</td>\n",
       "      <td>chrX:2781927:A:G</td>\n",
       "      <td>rs2306737</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.053882</td>\n",
       "      <td>0.057451</td>\n",
       "      <td>1.055360</td>\n",
       "      <td>0.942968</td>\n",
       "      <td>...</td>\n",
       "      <td>0.348336</td>\n",
       "      <td>0.436336</td>\n",
       "      <td>0.455169</td>\n",
       "      <td>0.427670</td>\n",
       "      <td>2625</td>\n",
       "      <td>863</td>\n",
       "      <td>1762</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X</td>\n",
       "      <td>2781986</td>\n",
       "      <td>chrX:2781986:T:C</td>\n",
       "      <td>rs2306736</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.068425</td>\n",
       "      <td>0.057451</td>\n",
       "      <td>1.070820</td>\n",
       "      <td>0.956783</td>\n",
       "      <td>...</td>\n",
       "      <td>0.233619</td>\n",
       "      <td>0.437334</td>\n",
       "      <td>0.457806</td>\n",
       "      <td>0.427913</td>\n",
       "      <td>2631</td>\n",
       "      <td>868</td>\n",
       "      <td>1763</td>\n",
       "      <td>6016</td>\n",
       "      <td>1896</td>\n",
       "      <td>4120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS                ID        rsID EffectAllele OtherAllele  \\\n",
       "0     X  2781514  chrX:2781514:C:A    rs311165            A           C   \n",
       "1     X  2781604  chrX:2781604:G:T  rs28579419            T           G   \n",
       "2     X  2781635  chrX:2781635:G:A  rs60075487            A           G   \n",
       "3     X  2781927  chrX:2781927:A:G   rs2306737            A           G   \n",
       "4     X  2781986  chrX:2781986:T:C   rs2306736            T           C   \n",
       "\n",
       "       BETA   BETA_SE        OR  OR_low95  ...         P  MAF_EffectAllele  \\\n",
       "0 -0.025560  0.058254  0.974764  0.869586  ...  0.660837          0.416057   \n",
       "1  0.022065  0.082083  1.022310  0.870387  ...  0.788093          0.138630   \n",
       "2  0.114979  0.077130  1.121850  0.964451  ...  0.136048          0.153757   \n",
       "3  0.053882  0.057451  1.055360  0.942968  ...  0.348336          0.436336   \n",
       "4  0.068425  0.057451  1.070820  0.956783  ...  0.233619          0.437334   \n",
       "\n",
       "   MAF_EffectAllele_CASE  MAF_EffectAllele_CTRL  CT_EffectAllele  \\\n",
       "0               0.410338               0.418689             2503   \n",
       "1               0.143460               0.136408              834   \n",
       "2               0.165084               0.148544              925   \n",
       "3               0.455169               0.427670             2625   \n",
       "4               0.457806               0.427913             2631   \n",
       "\n",
       "   CT_EffectAllele_CASE  CT_EffectAllele_CTRL  ALLELE_CT  CASE_ALLELE_CT  \\\n",
       "0                   778                  1725       6016            1896   \n",
       "1                   272                   562       6016            1896   \n",
       "2                   313                   612       6016            1896   \n",
       "3                   863                  1762       6016            1896   \n",
       "4                   868                  1763       6016            1896   \n",
       "\n",
       "   CTRL_ALLELE_CT  \n",
       "0            4120  \n",
       "1            4120  \n",
       "2            4120  \n",
       "3            4120  \n",
       "4            4120  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xwas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2074203, 15)\n",
      "(0, 16)\n",
      "(2074203, 15)\n",
      "(2086703, 15)\n",
      "(0, 16)\n",
      "(2086703, 15)\n",
      "(2090599, 15)\n",
      "(0, 16)\n",
      "(2090599, 15)\n",
      "(2085779, 15)\n",
      "(0, 16)\n",
      "(2085779, 15)\n",
      "(2087214, 15)\n",
      "(0, 16)\n",
      "(2087214, 15)\n",
      "(2086016, 15)\n",
      "(0, 16)\n",
      "(2086016, 15)\n",
      "(2096348, 15)\n",
      "(0, 16)\n",
      "(2096348, 15)\n",
      "(2052472, 15)\n",
      "(0, 16)\n",
      "(2052472, 15)\n",
      "(2166400, 15)\n",
      "(0, 16)\n",
      "(2166400, 15)\n",
      "(2133849, 15)\n",
      "(0, 16)\n",
      "(2133849, 15)\n",
      "(1997686, 15)\n",
      "(0, 16)\n",
      "(1997686, 15)\n",
      "(2091078, 15)\n",
      "(0, 16)\n",
      "(2091078, 15)\n",
      "(2022392, 15)\n",
      "(0, 16)\n",
      "(2022392, 15)\n"
     ]
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "brain_list = ['Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra']\n",
    "\n",
    "for i in brain_list:\n",
    "    input = \"/data/NDRS_LNG/GTEX_v8/GTEX_v8_chrX_brain/eQTL/\" + i + \".v8.EUR.allpairs.chrX.txt\"\n",
    "    eqtl = pd.read_csv(input,sep=\"\\t\")\n",
    "    \n",
    "    temp = eqtl['variant_id'].str.split(\"_\", expand=True)\n",
    "    temp.columns = ['chrom','POS','ref','alt','build']\n",
    "    temp.head()\n",
    "    \n",
    "    eqtl_updated = pd.concat([eqtl,temp[['chrom','POS','ref','alt']]], axis=1)\n",
    "    eqtl_updated['var_id'] = eqtl_updated['chrom'] + \":\" + eqtl_updated['POS'] + \":\" + eqtl_updated['ref'] + \":\" + eqtl_updated['alt']\n",
    "    eqtl_updated['var_id_alt'] = eqtl_updated['chrom'] + \":\" + eqtl_updated['POS'] + \":\" + eqtl_updated['alt'] + \":\" + eqtl_updated['ref']\n",
    "    \n",
    "    # check against xwas summstats SNP ID\n",
    "    ## eqtl_updated var_id that matches chrom:pos:ref:alt\n",
    "    temp1 = eqtl_updated[(eqtl_updated.var_id.isin(xwas.ID))]\n",
    "    temp1['ID'] = temp1['var_id']\n",
    "    temp1['REF'] = temp1['ref']\n",
    "    temp1['ALT'] = temp1['alt']\n",
    "    temp1 = temp1[['phenotype_id', 'variant_id', 'tss_distance', 'ma_samples', 'ma_count',\n",
    "       'pval_nominal', 'beta', 'varbeta', 'N', 'MAF', 'chrom', 'POS', 'REF',\n",
    "       'ALT', 'ID']]\n",
    "    print(temp1.shape)\n",
    "\n",
    "    ## eqtl_updated var_id_alt that matches chrom:pos:ref:alt\n",
    "    # for those that matches, will need to flip the beta and MAF values\n",
    "    temp2 = eqtl_updated[(eqtl_updated.var_id_alt.isin(xwas.ID))]\n",
    "    print(temp2.shape)\n",
    "    temp2['beta_corr'] = -1 * temp2['beta']\n",
    "    temp2['MAF_corr'] = 0.5 - temp2['MAF']\n",
    "    temp2['ID'] = temp2['var_id_alt']\n",
    "    temp2['REF'] = temp2['alt']\n",
    "    temp2['ALT'] = temp2['ref']\n",
    "    temp2 = temp2[['phenotype_id', 'variant_id', 'tss_distance', 'ma_samples', 'ma_count',\n",
    "       'pval_nominal', 'beta_corr', 'varbeta', 'N', 'MAF_corr', 'chrom', 'POS', 'REF',\n",
    "       'ALT', 'ID']]\n",
    "    temp2 = temp2.rename(columns={'beta_corr':'beta', 'MAF_corr':'MAF'})\n",
    "\n",
    "    # concat both temp1 and temp2 dataframes - these will only contain variants that are shared between xwas summ stats and eqtl\n",
    "    eqtl_updated_clean = pd.concat([temp1,temp2], axis = 0)\n",
    "    print(eqtl_updated_clean.shape)\n",
    "    \n",
    "    # get rsid from xwas summ stats\n",
    "    rsid_map = xwas[['ID','rsID']]\n",
    "    eqtl_updated_clean_rsid = eqtl_updated_clean.merge(rsid_map, on=\"ID\")\n",
    "    eqtl_updated_clean_rsid.to_csv(\"Analysis.COLOC_redo/GTEXv8_eQTL_brains/\" + i + \".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\\t\",index=False,header=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5566, 15)\n",
      "(0, 16)\n",
      "(5566, 15)\n",
      "(7962, 15)\n",
      "(0, 16)\n",
      "(7962, 15)\n",
      "(16002, 15)\n",
      "(0, 16)\n",
      "(16002, 15)\n",
      "(19720, 15)\n",
      "(0, 16)\n",
      "(19720, 15)\n",
      "(27471, 15)\n",
      "(0, 16)\n",
      "(27471, 15)\n",
      "(18755, 15)\n",
      "(0, 16)\n",
      "(18755, 15)\n",
      "(17151, 15)\n",
      "(0, 16)\n",
      "(17151, 15)\n",
      "(8407, 15)\n",
      "(0, 16)\n",
      "(8407, 15)\n",
      "(10050, 15)\n",
      "(0, 16)\n",
      "(10050, 15)\n",
      "(16032, 15)\n",
      "(0, 16)\n",
      "(16032, 15)\n",
      "(10713, 15)\n",
      "(0, 16)\n",
      "(10713, 15)\n",
      "(8080, 15)\n",
      "(0, 16)\n",
      "(8080, 15)\n",
      "(3999, 15)\n",
      "(0, 16)\n",
      "(3999, 15)\n"
     ]
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "brain_list = ['Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra']\n",
    "\n",
    "for i in brain_list:\n",
    "    input = \"/data/NDRS_LNG/GTEX_v8/GTEX_v8_chrX_brain/eQTL/\" + i + \".v8.EUR.signif_pairs.txt\"\n",
    "    eqtl = pd.read_csv(input,sep=\"\\t\")\n",
    "    \n",
    "    temp = eqtl['variant_id'].str.split(\"_\", expand=True)\n",
    "    temp.columns = ['chrom','POS','ref','alt','build']\n",
    "    temp.head()\n",
    "    \n",
    "    eqtl_updated = pd.concat([eqtl,temp[['chrom','POS','ref','alt']]], axis=1)\n",
    "    eqtl_updated['var_id'] = eqtl_updated['chrom'] + \":\" + eqtl_updated['POS'] + \":\" + eqtl_updated['ref'] + \":\" + eqtl_updated['alt']\n",
    "    eqtl_updated['var_id_alt'] = eqtl_updated['chrom'] + \":\" + eqtl_updated['POS'] + \":\" + eqtl_updated['alt'] + \":\" + eqtl_updated['ref']\n",
    "    \n",
    "    # check against xwas summstats SNP ID\n",
    "    ## eqtl_updated var_id that matches chrom:pos:ref:alt\n",
    "    temp1 = eqtl_updated[(eqtl_updated.var_id.isin(xwas.ID))]\n",
    "    temp1['ID'] = temp1['var_id']\n",
    "    temp1['REF'] = temp1['ref']\n",
    "    temp1['ALT'] = temp1['alt']\n",
    "    temp1 = temp1[['phenotype_id', 'variant_id', 'tss_distance', 'ma_samples', 'ma_count',\n",
    "       'pval_nominal', 'beta', 'varbeta', 'N', 'MAF', 'chrom', 'POS', 'REF',\n",
    "       'ALT', 'ID']]\n",
    "    print(temp1.shape)\n",
    "\n",
    "    ## eqtl_updated var_id_alt that matches chrom:pos:ref:alt\n",
    "    # for those that matches, will need to flip the beta and MAF values\n",
    "    temp2 = eqtl_updated[(eqtl_updated.var_id_alt.isin(xwas.ID))]\n",
    "    print(temp2.shape)\n",
    "    temp2['beta_corr'] = -1 * temp2['beta']\n",
    "    temp2['MAF_corr'] = 0.5 - temp2['MAF']\n",
    "    temp2['ID'] = temp2['var_id_alt']\n",
    "    temp2['REF'] = temp2['alt']\n",
    "    temp2['ALT'] = temp2['ref']\n",
    "    temp2 = temp2[['phenotype_id', 'variant_id', 'tss_distance', 'ma_samples', 'ma_count',\n",
    "       'pval_nominal', 'beta_corr', 'varbeta', 'N', 'MAF_corr', 'chrom', 'POS', 'REF',\n",
    "       'ALT', 'ID']]\n",
    "    temp2 = temp2.rename(columns={'beta_corr':'beta', 'MAF_corr':'MAF'})\n",
    "\n",
    "    # concat both temp1 and temp2 dataframes - these will only contain variants that are shared between xwas summ stats and eqtl\n",
    "    eqtl_updated_clean = pd.concat([temp1,temp2], axis = 0)\n",
    "    print(eqtl_updated_clean.shape)\n",
    "    \n",
    "    # get rsid from xwas summ stats\n",
    "    rsid_map = xwas[['ID','rsID']]\n",
    "    eqtl_updated_clean_rsid = eqtl_updated_clean.merge(rsid_map, on=\"ID\")\n",
    "    eqtl_updated_clean_rsid.to_csv(\"Analysis.COLOC_redo/GTEXv8_eQTL_brains/\" + i + \".v8.EUR.signif_pairs.chrX_harmonized_snpid_shared.txt\",sep=\"\\t\",index=False,header=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(675, 15)\n",
      "(0, 16)\n",
      "(675, 15)\n",
      "(688, 15)\n",
      "(0, 16)\n",
      "(688, 15)\n",
      "(679, 15)\n",
      "(0, 16)\n",
      "(679, 15)\n",
      "(687, 15)\n",
      "(0, 16)\n",
      "(687, 15)\n",
      "(693, 15)\n",
      "(0, 16)\n",
      "(693, 15)\n",
      "(686, 15)\n",
      "(0, 16)\n",
      "(686, 15)\n",
      "(683, 15)\n",
      "(0, 16)\n",
      "(683, 15)\n",
      "(680, 15)\n",
      "(0, 16)\n",
      "(680, 15)\n",
      "(704, 15)\n",
      "(0, 16)\n",
      "(704, 15)\n",
      "(690, 15)\n",
      "(0, 16)\n",
      "(690, 15)\n",
      "(635, 15)\n",
      "(0, 16)\n",
      "(635, 15)\n",
      "(698, 15)\n",
      "(0, 16)\n",
      "(698, 15)\n",
      "(652, 15)\n",
      "(0, 16)\n",
      "(652, 15)\n"
     ]
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "brain_list = ['Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra']\n",
    "\n",
    "for i in brain_list:\n",
    "    input = \"/data/NDRS_LNG/GTEX_v8/GTEX_v8_chrX_brain/eQTL/\" + i + \".v8.EUR.egenes.txt\"\n",
    "    eqtl = pd.read_csv(input,sep=\"\\t\")\n",
    "    \n",
    "    temp = eqtl['variant_id'].str.split(\"_\", expand=True)\n",
    "    temp.columns = ['chrom','POS','ref','alt','build']\n",
    "    temp.head()\n",
    "    \n",
    "    eqtl_updated = pd.concat([eqtl,temp[['chrom','POS','ref','alt']]], axis=1)\n",
    "    eqtl_updated['var_id'] = eqtl_updated['chrom'] + \":\" + eqtl_updated['POS'] + \":\" + eqtl_updated['ref'] + \":\" + eqtl_updated['alt']\n",
    "    eqtl_updated['var_id_alt'] = eqtl_updated['chrom'] + \":\" + eqtl_updated['POS'] + \":\" + eqtl_updated['alt'] + \":\" + eqtl_updated['ref']\n",
    "    \n",
    "    # check against xwas summstats SNP ID\n",
    "    ## eqtl_updated var_id that matches chrom:pos:ref:alt\n",
    "    temp1 = eqtl_updated[(eqtl_updated.var_id.isin(xwas.ID))]\n",
    "    temp1['ID'] = temp1['var_id']\n",
    "    temp1['REF'] = temp1['ref']\n",
    "    temp1['ALT'] = temp1['alt']\n",
    "    temp1 = temp1[['phenotype_id', 'variant_id', 'tss_distance', 'ma_samples', 'ma_count',\n",
    "       'pval_nominal', 'beta', 'varbeta', 'N', 'MAF', 'chrom', 'POS', 'REF',\n",
    "       'ALT', 'ID']]\n",
    "    print(temp1.shape)\n",
    "\n",
    "    ## eqtl_updated var_id_alt that matches chrom:pos:ref:alt\n",
    "    # for those that matches, will need to flip the beta and MAF values\n",
    "    temp2 = eqtl_updated[(eqtl_updated.var_id_alt.isin(xwas.ID))]\n",
    "    print(temp2.shape)\n",
    "    temp2['beta_corr'] = -1 * temp2['beta']\n",
    "    temp2['MAF_corr'] = 0.5 - temp2['MAF']\n",
    "    temp2['ID'] = temp2['var_id_alt']\n",
    "    temp2['REF'] = temp2['alt']\n",
    "    temp2['ALT'] = temp2['ref']\n",
    "    temp2 = temp2[['phenotype_id', 'variant_id', 'tss_distance', 'ma_samples', 'ma_count',\n",
    "       'pval_nominal', 'beta_corr', 'varbeta', 'N', 'MAF_corr', 'chrom', 'POS', 'REF',\n",
    "       'ALT', 'ID']]\n",
    "    temp2 = temp2.rename(columns={'beta_corr':'beta', 'MAF_corr':'MAF'})\n",
    "\n",
    "    # concat both temp1 and temp2 dataframes - these will only contain variants that are shared between xwas summ stats and eqtl\n",
    "    eqtl_updated_clean = pd.concat([temp1,temp2], axis = 0)\n",
    "    print(eqtl_updated_clean.shape)\n",
    "    \n",
    "    # get rsid from xwas summ stats\n",
    "    rsid_map = xwas[['ID','rsID']]\n",
    "    eqtl_updated_clean_rsid = eqtl_updated_clean.merge(rsid_map, on=\"ID\")\n",
    "    eqtl_updated_clean_rsid.to_csv(\"Analysis.COLOC_redo/GTEXv8_eQTL_brains/\" + i + \".v8.EUR.egenes.chrX_harmonized_snpid_shared.txt\",sep=\"\\t\",index=False,header=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### run COLOC for each brain tissue and for each subsetted xwas region\n",
    "\n",
    "#### Download LD binaries for creating LD matrix for analysis using `coloc.susie`\n",
    "\n",
    "ref link: `https://mrcieu.github.io/ieugwasr/reference/ld_matrix.html`\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/\n",
    "mkdir LD\n",
    "cd LD\n",
    "wget http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/LD\n",
    "tar -xvzf 1kg.v3.tgz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### females-only (unconditioned on APOE4)\n",
    "##### index_var1 (rs141773145)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[-] Unloading gcc  11.3.0  ... \n",
      "[-] Unloading HDF5  1.12.2 \n",
      "[-] Unloading netcdf  4.9.0 \n",
      "[-] Unloading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[-] Unloading pandoc  2.18  on cn3180 \n",
      "[-] Unloading R 4.3.0 \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading HDF5  1.12.2 \n",
      "[+] Loading netcdf  4.9.0 \n",
      "[-] Unloading gcc  11.3.0  ... \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[+] Loading pandoc  2.18  on cn3180 \n",
      "[+] Loading R 4.3.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.3.0 (2023-04-21) -- \"Already Tomorrow\"\n",
      "Copyright (C) 2023 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.4\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.4.4     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.3     ✔ tidyr     1.3.0\n",
      "✔ purrr     1.0.2     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::between()     masks data.table::between()\n",
      "✖ dplyr::filter()      masks stats::filter()\n",
      "✖ dplyr::first()       masks data.table::first()\n",
      "✖ lubridate::hour()    masks data.table::hour()\n",
      "✖ lubridate::isoweek() masks data.table::isoweek()\n",
      "✖ dplyr::lag()         masks stats::lag()\n",
      "✖ dplyr::last()        masks data.table::last()\n",
      "✖ lubridate::mday()    masks data.table::mday()\n",
      "✖ lubridate::minute()  masks data.table::minute()\n",
      "✖ lubridate::month()   masks data.table::month()\n",
      "✖ lubridate::quarter() masks data.table::quarter()\n",
      "✖ lubridate::second()  masks data.table::second()\n",
      "✖ purrr::transpose()   masks data.table::transpose()\n",
      "✖ lubridate::wday()    masks data.table::wday()\n",
      "✖ lubridate::week()    masks data.table::week()\n",
      "✖ lubridate::yday()    masks data.table::yday()\n",
      "✖ lubridate::year()    masks data.table::year()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> # install.packages(\"coloc\")\n",
      "> require(coloc)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: coloc\n",
      "This is coloc version 5.2.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
      "> require(ieugwasr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ieugwasr\n",
      "API: public: http://gwas-api.mrcieu.ac.uk/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
      "> genetics.binaRies::get_plink_binary()\n",
      "[1] \"/gpfs/gsfs9/users/chiarp/R/rhel8/4.3/genetics.binaRies/bin/plink\"\n",
      "> \n",
      "> brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
      "> \n",
      "> run_coloc <- function(rsid,i){\n",
      "+     # read in files\n",
      "+     tissue <- brain_tissues[i]\n",
      "+     xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     dim(eqtl)\n",
      "+     dim(xwas)    \n",
      "+     \n",
      "+     # prep so that only columns that are needed for coloc is present\n",
      "+     eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID) %>%\n",
      "+                    filter(!is.na(beta)) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
      "+                    arrange(snp,pval_nominal) %>%\n",
      "+                    group_by(snp) %>%\n",
      "+                    slice(1:1) %>%\n",
      "+                    arrange(position) %>%\n",
      "+                    data.frame()\n",
      "+   \n",
      "+     xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID, \n",
      "+                           beta = BETA,\n",
      "+                           varbeta = (BETA_SE**2)) %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta,P)\n",
      "+     \n",
      "+     # check dimensions\n",
      "+     dim(eqtl_subset)\n",
      "+     dim(xwas_subset)\n",
      "+         \n",
      "+     # convert to acceptable data structure for input in coloc\n",
      "+     eqtl_subset_list <- as.list(eqtl_subset)\n",
      "+     eqtl_subset_list$type <- \"quant\"\n",
      "+     xwas_subset_list <- as.list(xwas_subset)\n",
      "+     xwas_subset_list$type <- \"cc\"\n",
      "+     xwas_subset_list$N <- 2591 + 4023\n",
      "+     #check_dataset(eqtl_subset_list)\n",
      "+     #check_dataset(xwas_subset_list)\n",
      "+         \n",
      "+     # run coloc.abf\n",
      "+     print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
      "+     my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
      "+                     dataset2=eqtl_subset_list)\n",
      "+     print(my.res)\n",
      "+     output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
      "+ \n",
      "+     summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
      "+     priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     \n",
      "+     summary\n",
      "+     results\n",
      "+     priors\n",
      "+     \n",
      "+     write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
      "+ }\n",
      "> \n",
      "> for (i in 1:length(brain_tissues)){\n",
      "+     run_coloc(\"rs141773145\", i)\n",
      "+ }\n",
      "[1] \"run coloc for: Brain_Amygdala\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.6360    0.0981    0.2180    0.0336    0.0137 \n",
      "[1] \"PP abf for shared variant: 1.37%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5362679 0.2015 0.2015 0.0405821 0.02015\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.015000e+03 6.361878e-01 9.806622e-02 2.183740e-01 3.364789e-02 1.372406e-02 \n",
      "[1] \"saving coloc results for: Brain_Amygdala\"\n",
      "[1] \"run coloc for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.6440    0.0998    0.2090    0.0324    0.0138 \n",
      "[1] \"PP abf for shared variant: 1.38%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5345154 0.2022 0.2022 0.04086462 0.02022\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.022000e+03 6.444935e-01 9.980599e-02 2.094822e-01 3.242653e-02 1.379177e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "[1] \"run coloc for: Brain_Caudate_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.01330   0.00208   0.82600   0.12900   0.03020 \n",
      "[1] \"PP abf for shared variant: 3.02%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5315088 0.2034 0.2034 0.04135122 0.02034\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.034000e+03 1.333040e-02 2.079736e-03 8.256164e-01 1.287780e-01 3.019551e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Caudate_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Cerebellar_Hemisphere\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.06010   0.00935   0.78000   0.12100   0.02930 \n",
      "[1] \"PP abf for shared variant: 2.93%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5320101 0.2032 0.2032 0.04126992 0.02032\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.032000e+03 6.007769e-02 9.348435e-03 7.799209e-01 1.213309e-01 2.932206e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Cerebellar_Hemisphere\"\n",
      "[1] \"run coloc for: Brain_Cerebellum\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 5.89e-05  9.19e-06  8.39e-01  1.31e-01  2.98e-02 \n",
      "[1] \"PP abf for shared variant: 2.98%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5312581 0.2035 0.2035 0.0413919 0.02035\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.035000e+03 5.890564e-05 9.194100e-06 8.392040e-01 1.309547e-01 2.977322e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Cerebellum\"\n",
      "[1] \"run coloc for: Brain_Cortex\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.4600    0.0716    0.3900    0.0608    0.0181 \n",
      "[1] \"PP abf for shared variant: 1.81%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5312581 0.2035 0.2035 0.0413919 0.02035\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.035000e+03 4.595582e-01 7.163649e-02 3.899323e-01 6.076500e-02 1.810802e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Cortex\"\n",
      "[1] \"run coloc for: Brain_Frontal_Cortex_BA9\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1060    0.0164    0.7360    0.1140    0.0279 \n",
      "[1] \"PP abf for shared variant: 2.79%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5330124 0.2028 0.2028 0.04110756 0.02028\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.028000e+03 1.055174e-01 1.640405e-02 7.358249e-01 1.143657e-01 2.788800e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Frontal_Cortex_BA9\"\n",
      "[1] \"run coloc for: Brain_Hippocampus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.4700    0.0733    0.3790    0.0592    0.0185 \n",
      "[1] \"PP abf for shared variant: 1.85%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5312581 0.2035 0.2035 0.0413919 0.02035\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.035000e+03 4.695511e-01 7.328181e-02 3.794766e-01 5.920560e-02 1.848491e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Hippocampus\"\n",
      "[1] \"run coloc for: Brain_Hypothalamus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.3760    0.0583    0.4710    0.0730    0.0216 \n",
      "[1] \"PP abf for shared variant: 2.16%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "       H0     H1     H2       H3      H4\n",
      " 0.533764 0.2025 0.2025 0.040986 0.02025\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.025000e+03 3.759906e-01 5.828073e-02 4.710916e-01 7.300031e-02 2.163673e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Hypothalamus\"\n",
      "[1] \"run coloc for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1830    0.0286    0.6600    0.1030    0.0251 \n",
      "[1] \"PP abf for shared variant: 2.51%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5312581 0.2035 0.2035 0.0413919 0.02035\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.035000e+03 1.832167e-01 2.859889e-02 6.600424e-01 1.030030e-01 2.513896e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Putamen_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.04550   0.00706   0.79200   0.12300   0.03220 \n",
      "[1] \"PP abf for shared variant: 3.22%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "       H0     H1     H2         H3      H4\n",
      " 0.533263 0.2027 0.2027 0.04106702 0.02027\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.027000e+03 4.546125e-02 7.064732e-03 7.922128e-01 1.230786e-01 3.218260e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Putamen_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Spinal_cord_cervical_c-1\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1470    0.0227    0.6950    0.1070    0.0279 \n",
      "[1] \"PP abf for shared variant: 2.79%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5355169 0.2018 0.2018 0.04070306 0.02018\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.018000e+03 1.467410e-01 2.265066e-02 6.953791e-01 1.073095e-01 2.791976e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Spinal_cord_cervical_c-1\"\n",
      "[1] \"run coloc for: Brain_Substantia_nigra\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.6540    0.1000    0.2010    0.0308    0.0140 \n",
      "[1] \"PP abf for shared variant: 1.4%\"\n",
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5385197 0.2006 0.2006 0.0402203 0.02006\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.006000e+03 6.537681e-01 1.000758e-01 2.013055e-01 3.080086e-02 1.404979e-02 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"saving coloc results for: Brain_Substantia_nigra\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 29 warnings (use warnings() to see them)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/females_only\n",
    "\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "# install.packages(\"coloc\")\n",
    "require(coloc)\n",
    "#devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
    "require(ieugwasr)\n",
    "#devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
    "genetics.binaRies::get_plink_binary()\n",
    "\n",
    "brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
    "\n",
    "run_coloc <- function(rsid,i){\n",
    "    # read in files\n",
    "    tissue <- brain_tissues[i]\n",
    "    xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    dim(eqtl)\n",
    "    dim(xwas)    \n",
    "    \n",
    "    # prep so that only columns that are needed for coloc is present\n",
    "    eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID) %>%\n",
    "                   filter(!is.na(beta)) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
    "                   arrange(snp,pval_nominal) %>%\n",
    "                   group_by(snp) %>%\n",
    "                   slice(1:1) %>%\n",
    "                   arrange(position) %>%\n",
    "                   data.frame()\n",
    "  \n",
    "    xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID, \n",
    "                          beta = BETA,\n",
    "                          varbeta = (BETA_SE**2)) %>%\n",
    "                   select(rsID, snp, position, beta, varbeta,P)\n",
    "    \n",
    "    # check dimensions\n",
    "    dim(eqtl_subset)\n",
    "    dim(xwas_subset)\n",
    "        \n",
    "    # convert to acceptable data structure for input in coloc\n",
    "    eqtl_subset_list <- as.list(eqtl_subset)\n",
    "    eqtl_subset_list$type <- \"quant\"\n",
    "    xwas_subset_list <- as.list(xwas_subset)\n",
    "    xwas_subset_list$type <- \"cc\"\n",
    "    xwas_subset_list$N <- 2591 + 4023\n",
    "    #check_dataset(eqtl_subset_list)\n",
    "    #check_dataset(xwas_subset_list)\n",
    "        \n",
    "    # run coloc.abf\n",
    "    print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
    "    my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
    "                    dataset2=eqtl_subset_list)\n",
    "    print(my.res)\n",
    "    output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
    "\n",
    "    summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
    "    priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    \n",
    "    summary\n",
    "    results\n",
    "    priors\n",
    "    \n",
    "    write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
    "}\n",
    "\n",
    "for (i in 1:length(brain_tissues)){\n",
    "    run_coloc(\"rs141773145\", i)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### index_var2 (rs12860838)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[-] Unloading gcc  11.3.0  ... \n",
      "[-] Unloading HDF5  1.12.2 \n",
      "[-] Unloading netcdf  4.9.0 \n",
      "[-] Unloading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[-] Unloading pandoc  2.18  on cn3180 \n",
      "[-] Unloading R 4.3.0 \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading HDF5  1.12.2 \n",
      "[+] Loading netcdf  4.9.0 \n",
      "[-] Unloading gcc  11.3.0  ... \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[+] Loading pandoc  2.18  on cn3180 \n",
      "[+] Loading R 4.3.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.3.0 (2023-04-21) -- \"Already Tomorrow\"\n",
      "Copyright (C) 2023 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.4\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.4.4     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.3     ✔ tidyr     1.3.0\n",
      "✔ purrr     1.0.2     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::between()     masks data.table::between()\n",
      "✖ dplyr::filter()      masks stats::filter()\n",
      "✖ dplyr::first()       masks data.table::first()\n",
      "✖ lubridate::hour()    masks data.table::hour()\n",
      "✖ lubridate::isoweek() masks data.table::isoweek()\n",
      "✖ dplyr::lag()         masks stats::lag()\n",
      "✖ dplyr::last()        masks data.table::last()\n",
      "✖ lubridate::mday()    masks data.table::mday()\n",
      "✖ lubridate::minute()  masks data.table::minute()\n",
      "✖ lubridate::month()   masks data.table::month()\n",
      "✖ lubridate::quarter() masks data.table::quarter()\n",
      "✖ lubridate::second()  masks data.table::second()\n",
      "✖ purrr::transpose()   masks data.table::transpose()\n",
      "✖ lubridate::wday()    masks data.table::wday()\n",
      "✖ lubridate::week()    masks data.table::week()\n",
      "✖ lubridate::yday()    masks data.table::yday()\n",
      "✖ lubridate::year()    masks data.table::year()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> # install.packages(\"coloc\")\n",
      "> require(coloc)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: coloc\n",
      "This is coloc version 5.2.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
      "> require(ieugwasr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ieugwasr\n",
      "API: public: http://gwas-api.mrcieu.ac.uk/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
      "> genetics.binaRies::get_plink_binary()\n",
      "[1] \"/gpfs/gsfs9/users/chiarp/R/rhel8/4.3/genetics.binaRies/bin/plink\"\n",
      "> \n",
      "> brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
      "> \n",
      "> run_coloc <- function(rsid,i){\n",
      "+     # read in files\n",
      "+     tissue <- brain_tissues[i]\n",
      "+     xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     dim(eqtl)\n",
      "+     dim(xwas)    \n",
      "+     \n",
      "+     # prep so that only columns that are needed for coloc is present\n",
      "+     eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID) %>%\n",
      "+                    filter(!is.na(beta)) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
      "+                    arrange(snp,pval_nominal) %>%\n",
      "+                    group_by(snp) %>%\n",
      "+                    slice(1:1) %>%\n",
      "+                    arrange(position) %>%\n",
      "+                    data.frame()\n",
      "+   \n",
      "+     xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID, \n",
      "+                           beta = BETA,\n",
      "+                           varbeta = (BETA_SE**2)) %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta,P)\n",
      "+     \n",
      "+     # check dimensions\n",
      "+     dim(eqtl_subset)\n",
      "+     dim(xwas_subset)\n",
      "+         \n",
      "+     # convert to acceptable data structure for input in coloc\n",
      "+     eqtl_subset_list <- as.list(eqtl_subset)\n",
      "+     eqtl_subset_list$type <- \"quant\"\n",
      "+     xwas_subset_list <- as.list(xwas_subset)\n",
      "+     xwas_subset_list$type <- \"cc\"\n",
      "+     xwas_subset_list$N <- 2591 + 4023\n",
      "+     #check_dataset(eqtl_subset_list)\n",
      "+     #check_dataset(xwas_subset_list)\n",
      "+         \n",
      "+     # run coloc.abf\n",
      "+     print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
      "+     my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
      "+                     dataset2=eqtl_subset_list)\n",
      "+     print(my.res)\n",
      "+     output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
      "+ \n",
      "+     summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
      "+     priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     \n",
      "+     write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
      "+ }\n",
      "> \n",
      "> for (i in 1:length(brain_tissues)){\n",
      "+     run_coloc(\"rs12860838\", i)\n",
      "+ }\n",
      "[1] \"run coloc for: Brain_Amygdala\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2390    0.3480    0.1520    0.2220    0.0383 \n",
      "[1] \"PP abf for shared variant: 3.83%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2564402 0.4859 0.4859 0.2360502 0.04859\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.859000e+03 2.394184e-01 3.484568e-01 1.522586e-01 2.215635e-01 3.830275e-02 \n",
      "[1] \"saving coloc results for: Brain_Amygdala\"\n",
      "[1] \"run coloc for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2140    0.3120    0.1790    0.2610    0.0353 \n",
      "[1] \"PP abf for shared variant: 3.53%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2665877 0.4892 0.4892 0.2392677 0.04892\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4892.0000000    0.2135729    0.3115344    0.1788009    0.2607779    0.0353139 \n",
      "[1] \"saving coloc results for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "[1] \"run coloc for: Brain_Caudate_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2230    0.3250    0.1660    0.2420    0.0446 \n",
      "[1] \"PP abf for shared variant: 4.46%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0   H1   H2       H3    H4\n",
      " -0.269051 0.49 0.49 0.240051 0.049\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4900.0000000    0.2227690    0.3251229    0.1656927    0.2417777    0.0446377 \n",
      "[1] \"saving coloc results for: Brain_Caudate_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Cerebellar_Hemisphere\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0241    0.0351    0.3580    0.5220    0.0615 \n",
      "[1] \"PP abf for shared variant: 6.15%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2681271 0.4897 0.4897 0.2397571 0.04897\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.897000e+03 2.405103e-02 3.510447e-02 3.575341e-01 5.217890e-01 6.152143e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellar_Hemisphere\"\n",
      "[1] \"run coloc for: Brain_Cerebellum\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1520    0.2220    0.2340    0.3410    0.0513 \n",
      "[1] \"PP abf for shared variant: 5.13%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.269667 0.4902 0.4902 0.240247 0.04902\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.902000e+03 1.520473e-01 2.219784e-01 2.336369e-01 3.410424e-01 5.129505e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellum\"\n",
      "[1] \"run coloc for: Brain_Cortex\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1950    0.2850    0.1750    0.2560    0.0891 \n",
      "[1] \"PP abf for shared variant: 8.91%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2684351 0.4898 0.4898 0.2398551 0.04898\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.898000e+03 1.949325e-01 2.845336e-01 1.754239e-01 2.559687e-01 8.914125e-02 \n",
      "[1] \"saving coloc results for: Brain_Cortex\"\n",
      "[1] \"run coloc for: Brain_Frontal_Cortex_BA9\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2370    0.3460    0.1540    0.2250    0.0391 \n",
      "[1] \"PP abf for shared variant: 3.91%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.269667 0.4902 0.4902 0.240247 0.04902\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4902.0000000    0.2367534    0.3456425    0.1538869    0.2246244    0.0390928 \n",
      "[1] \"saving coloc results for: Brain_Frontal_Cortex_BA9\"\n",
      "[1] \"run coloc for: Brain_Hippocampus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "    0.197     0.287     0.194     0.283     0.040 \n",
      "[1] \"PP abf for shared variant: 4%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.269667 0.4902 0.4902 0.240247 0.04902\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.902000e+03 1.966987e-01 2.871731e-01 1.935820e-01 2.825828e-01 3.996337e-02 \n",
      "[1] \"saving coloc results for: Brain_Hippocampus\"\n",
      "[1] \"run coloc for: Brain_Hypothalamus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "    0.239     0.348     0.152     0.222     0.039 \n",
      "[1] \"PP abf for shared variant: 3.9%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2672034 0.4894 0.4894 0.2394634 0.04894\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.894000e+03 2.385011e-01 3.479759e-01 1.523183e-01 2.221951e-01 3.900965e-02 \n",
      "[1] \"saving coloc results for: Brain_Hypothalamus\"\n",
      "[1] \"run coloc for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1950    0.2850    0.1920    0.2800    0.0485 \n",
      "[1] \"PP abf for shared variant: 4.85%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2702831 0.4904 0.4904 0.2404431 0.04904\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.904000e+03 1.951241e-01 2.849101e-01 1.916407e-01 2.797753e-01 4.854981e-02 \n",
      "[1] \"saving coloc results for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Putamen_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2120    0.3100    0.1720    0.2510    0.0545 \n",
      "[1] \"PP abf for shared variant: 5.45%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2665877 0.4892 0.4892 0.2392677 0.04892\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.892000e+03 2.124685e-01 3.099918e-01 1.720590e-01 2.509798e-01 5.450096e-02 \n",
      "[1] \"saving coloc results for: Brain_Putamen_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Spinal_cord_cervical_c-1\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2340    0.3410    0.1590    0.2320    0.0331 \n",
      "[1] \"PP abf for shared variant: 3.31%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2573618 0.4862 0.4862 0.2363418 0.04862\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.862000e+03 2.342308e-01 3.410173e-01 1.594840e-01 2.321601e-01 3.310783e-02 \n",
      "[1] \"saving coloc results for: Brain_Spinal_cord_cervical_c-1\"\n",
      "[1] \"run coloc for: Brain_Substantia_nigra\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "    0.253     0.362     0.135     0.193     0.057 \n",
      "[1] \"PP abf for shared variant: 5.7%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.1139816 0.4388 0.4388 0.1925016 0.04388\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.388000e+03 2.532806e-01 3.617263e-01 1.351027e-01 1.928918e-01 5.699862e-02 \n",
      "[1] \"saving coloc results for: Brain_Substantia_nigra\"\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 38 warnings (use warnings() to see them)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/females_only\n",
    "\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "# install.packages(\"coloc\")\n",
    "require(coloc)\n",
    "#devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
    "require(ieugwasr)\n",
    "#devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
    "genetics.binaRies::get_plink_binary()\n",
    "\n",
    "brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
    "\n",
    "run_coloc <- function(rsid,i){\n",
    "    # read in files\n",
    "    tissue <- brain_tissues[i]\n",
    "    xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    dim(eqtl)\n",
    "    dim(xwas)    \n",
    "    \n",
    "    # prep so that only columns that are needed for coloc is present\n",
    "    eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID) %>%\n",
    "                   filter(!is.na(beta)) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
    "                   arrange(snp,pval_nominal) %>%\n",
    "                   group_by(snp) %>%\n",
    "                   slice(1:1) %>%\n",
    "                   arrange(position) %>%\n",
    "                   data.frame()\n",
    "  \n",
    "    xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID, \n",
    "                          beta = BETA,\n",
    "                          varbeta = (BETA_SE**2)) %>%\n",
    "                   select(rsID, snp, position, beta, varbeta,P)\n",
    "    \n",
    "    # check dimensions\n",
    "    dim(eqtl_subset)\n",
    "    dim(xwas_subset)\n",
    "        \n",
    "    # convert to acceptable data structure for input in coloc\n",
    "    eqtl_subset_list <- as.list(eqtl_subset)\n",
    "    eqtl_subset_list$type <- \"quant\"\n",
    "    xwas_subset_list <- as.list(xwas_subset)\n",
    "    xwas_subset_list$type <- \"cc\"\n",
    "    xwas_subset_list$N <- 2591 + 4023\n",
    "    #check_dataset(eqtl_subset_list)\n",
    "    #check_dataset(xwas_subset_list)\n",
    "        \n",
    "    # run coloc.abf\n",
    "    print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
    "    my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
    "                    dataset2=eqtl_subset_list)\n",
    "    print(my.res)\n",
    "    output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
    "\n",
    "    summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
    "    priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    \n",
    "    write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
    "}\n",
    "\n",
    "for (i in 1:length(brain_tissues)){\n",
    "    run_coloc(\"rs12860838\", i)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### females-only (conditioned on APOE4)\n",
    "##### index_var1 (rs141773145)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[-] Unloading gcc  11.3.0  ... \n",
      "[-] Unloading HDF5  1.12.2 \n",
      "[-] Unloading netcdf  4.9.0 \n",
      "[-] Unloading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[-] Unloading pandoc  2.18  on cn3180 \n",
      "[-] Unloading R 4.3.0 \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading HDF5  1.12.2 \n",
      "[+] Loading netcdf  4.9.0 \n",
      "[-] Unloading gcc  11.3.0  ... \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[+] Loading pandoc  2.18  on cn3180 \n",
      "[+] Loading R 4.3.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.3.0 (2023-04-21) -- \"Already Tomorrow\"\n",
      "Copyright (C) 2023 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.4\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.4.4     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.3     ✔ tidyr     1.3.0\n",
      "✔ purrr     1.0.2     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::between()     masks data.table::between()\n",
      "✖ dplyr::filter()      masks stats::filter()\n",
      "✖ dplyr::first()       masks data.table::first()\n",
      "✖ lubridate::hour()    masks data.table::hour()\n",
      "✖ lubridate::isoweek() masks data.table::isoweek()\n",
      "✖ dplyr::lag()         masks stats::lag()\n",
      "✖ dplyr::last()        masks data.table::last()\n",
      "✖ lubridate::mday()    masks data.table::mday()\n",
      "✖ lubridate::minute()  masks data.table::minute()\n",
      "✖ lubridate::month()   masks data.table::month()\n",
      "✖ lubridate::quarter() masks data.table::quarter()\n",
      "✖ lubridate::second()  masks data.table::second()\n",
      "✖ purrr::transpose()   masks data.table::transpose()\n",
      "✖ lubridate::wday()    masks data.table::wday()\n",
      "✖ lubridate::week()    masks data.table::week()\n",
      "✖ lubridate::yday()    masks data.table::yday()\n",
      "✖ lubridate::year()    masks data.table::year()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> # install.packages(\"coloc\")\n",
      "> require(coloc)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: coloc\n",
      "This is coloc version 5.2.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
      "> require(ieugwasr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ieugwasr\n",
      "API: public: http://gwas-api.mrcieu.ac.uk/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
      "> genetics.binaRies::get_plink_binary()\n",
      "[1] \"/gpfs/gsfs9/users/chiarp/R/rhel8/4.3/genetics.binaRies/bin/plink\"\n",
      "> \n",
      "> brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
      "> \n",
      "> run_coloc <- function(rsid,i){\n",
      "+     # read in files\n",
      "+     tissue <- brain_tissues[i]\n",
      "+     xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     dim(eqtl)\n",
      "+     dim(xwas)    \n",
      "+     \n",
      "+     # prep so that only columns that are needed for coloc is present\n",
      "+     eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID) %>%\n",
      "+                    filter(!is.na(beta)) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
      "+                    arrange(snp,pval_nominal) %>%\n",
      "+                    group_by(snp) %>%\n",
      "+                    slice(1:1) %>%\n",
      "+                    arrange(position) %>%\n",
      "+                    data.frame()\n",
      "+   \n",
      "+     xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID, \n",
      "+                           beta = BETA,\n",
      "+                           varbeta = (BETA_SE**2)) %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta,P)\n",
      "+     \n",
      "+     # check dimensions\n",
      "+     dim(eqtl_subset)\n",
      "+     dim(xwas_subset)\n",
      "+         \n",
      "+     # convert to acceptable data structure for input in coloc\n",
      "+     eqtl_subset_list <- as.list(eqtl_subset)\n",
      "+     eqtl_subset_list$type <- \"quant\"\n",
      "+     xwas_subset_list <- as.list(xwas_subset)\n",
      "+     xwas_subset_list$type <- \"cc\"\n",
      "+     xwas_subset_list$N <- 2591 + 4023\n",
      "+     #check_dataset(eqtl_subset_list)\n",
      "+     #check_dataset(xwas_subset_list)\n",
      "+         \n",
      "+     # run coloc.abf\n",
      "+     print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
      "+     my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
      "+                     dataset2=eqtl_subset_list)\n",
      "+     print(my.res)\n",
      "+     output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
      "+ \n",
      "+     summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
      "+     priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     \n",
      "+     write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
      "+ }\n",
      "> \n",
      "> for (i in 1:length(brain_tissues)){\n",
      "+     run_coloc(\"rs141773145\", i)\n",
      "+ }\n",
      "[1] \"run coloc for: Brain_Amygdala\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.6280    0.1060    0.2150    0.0362    0.0148 \n",
      "[1] \"PP abf for shared variant: 1.48%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5367684 0.2013 0.2013 0.04050156 0.02013\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.013000e+03 6.278530e-01 1.057056e-01 2.153844e-01 3.624741e-02 1.480954e-02 \n",
      "[1] \"saving coloc results for: Brain_Amygdala\"\n",
      "[1] \"run coloc for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.6360    0.1080    0.2070    0.0349    0.0148 \n",
      "[1] \"PP abf for shared variant: 1.48%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5345154 0.2022 0.2022 0.04086462 0.02022\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.022000e+03 6.359438e-01 1.075629e-01 2.067032e-01 3.494672e-02 1.484332e-02 \n",
      "[1] \"saving coloc results for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "[1] \"run coloc for: Brain_Caudate_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.01310   0.00224   0.81400   0.13900   0.03170 \n",
      "[1] \"PP abf for shared variant: 3.17%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5320101 0.2032 0.2032 0.04126992 0.02032\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.032000e+03 1.314893e-02 2.237809e-03 8.143719e-01 1.385658e-01 3.167561e-02 \n",
      "[1] \"saving coloc results for: Brain_Caudate_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Cerebellar_Hemisphere\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0593    0.0101    0.7700    0.1310    0.0302 \n",
      "[1] \"PP abf for shared variant: 3.02%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0    H1    H2        H3     H4\n",
      " 0.5325113 0.203 0.203 0.0411887 0.0203\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.030000e+03 5.929509e-02 1.007064e-02 7.697602e-01 1.307054e-01 3.016871e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellar_Hemisphere\"\n",
      "[1] \"run coloc for: Brain_Cerebellum\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 5.81e-05  9.89e-06  8.27e-01  1.41e-01  3.17e-02 \n",
      "[1] \"PP abf for shared variant: 3.17%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5317594 0.2033 0.2033 0.04131056 0.02033\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.033000e+03 5.807942e-05 9.888277e-06 8.274256e-01 1.408412e-01 3.166530e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellum\"\n",
      "[1] \"run coloc for: Brain_Cortex\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.4530    0.0771    0.3850    0.0654    0.0193 \n",
      "[1] \"PP abf for shared variant: 1.93%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5317594 0.2033 0.2033 0.04131056 0.02033\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.033000e+03 4.534575e-01 7.714125e-02 3.846618e-01 6.541856e-02 1.932089e-02 \n",
      "[1] \"saving coloc results for: Brain_Cortex\"\n",
      "[1] \"run coloc for: Brain_Frontal_Cortex_BA9\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1040    0.0177    0.7270    0.1230    0.0282 \n",
      "[1] \"PP abf for shared variant: 2.82%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5335135 0.2026 0.2026 0.0410265 0.02026\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.026000e+03 1.042159e-01 1.767424e-02 7.267259e-01 1.232191e-01 2.816486e-02 \n",
      "[1] \"saving coloc results for: Brain_Frontal_Cortex_BA9\"\n",
      "[1] \"run coloc for: Brain_Hippocampus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.4640    0.0789    0.3750    0.0637    0.0193 \n",
      "[1] \"PP abf for shared variant: 1.93%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5317594 0.2033 0.2033 0.04131056 0.02033\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.033000e+03 4.635490e-01 7.890686e-02 3.745293e-01 6.373434e-02 1.928045e-02 \n",
      "[1] \"saving coloc results for: Brain_Hippocampus\"\n",
      "[1] \"run coloc for: Brain_Hypothalamus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.3710    0.0628    0.4650    0.0787    0.0224 \n",
      "[1] \"PP abf for shared variant: 2.24%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5340145 0.2024 0.2024 0.04094552 0.02024\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.024000e+03 3.711261e-01 6.283345e-02 4.649554e-01 7.869681e-02 2.238822e-02 \n",
      "[1] \"saving coloc results for: Brain_Hypothalamus\"\n",
      "[1] \"run coloc for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1810    0.0308    0.6510    0.1110    0.0262 \n",
      "[1] \"PP abf for shared variant: 2.62%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5317594 0.2033 0.2033 0.04131056 0.02033\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.033000e+03 1.808113e-01 3.078467e-02 6.513384e-01 1.108697e-01 2.619584e-02 \n",
      "[1] \"saving coloc results for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Putamen_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0448    0.0076    0.7810    0.1320    0.0340 \n",
      "[1] \"PP abf for shared variant: 3.4%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5335135 0.2026 0.2026 0.0410265 0.02026\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.026000e+03 4.482508e-02 7.604767e-03 7.811161e-01 1.324857e-01 3.396828e-02 \n",
      "[1] \"saving coloc results for: Brain_Putamen_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Spinal_cord_cervical_c-1\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1450    0.0244    0.6870    0.1160    0.0279 \n",
      "[1] \"PP abf for shared variant: 2.79%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.5357673 0.2017 0.2017 0.04066272 0.02017\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.017000e+03 1.449731e-01 2.442240e-02 6.869839e-01 1.157025e-01 2.791806e-02 \n",
      "[1] \"saving coloc results for: Brain_Spinal_cord_cervical_c-1\"\n",
      "[1] \"run coloc for: Brain_Substantia_nigra\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.6450    0.1080    0.1980    0.0332    0.0157 \n",
      "[1] \"PP abf for shared variant: 1.57%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.5387698 0.2005 0.2005 0.0401802 0.02005\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.005000e+03 6.447120e-01 1.079086e-01 1.984477e-01 3.319942e-02 1.573231e-02 \n",
      "[1] \"saving coloc results for: Brain_Substantia_nigra\"\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 29 warnings (use warnings() to see them)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/females_only_cond_ApoE4/\n",
    "\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "# install.packages(\"coloc\")\n",
    "require(coloc)\n",
    "#devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
    "require(ieugwasr)\n",
    "#devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
    "genetics.binaRies::get_plink_binary()\n",
    "\n",
    "brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
    "\n",
    "run_coloc <- function(rsid,i){\n",
    "    # read in files\n",
    "    tissue <- brain_tissues[i]\n",
    "    xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    dim(eqtl)\n",
    "    dim(xwas)    \n",
    "    \n",
    "    # prep so that only columns that are needed for coloc is present\n",
    "    eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID) %>%\n",
    "                   filter(!is.na(beta)) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
    "                   arrange(snp,pval_nominal) %>%\n",
    "                   group_by(snp) %>%\n",
    "                   slice(1:1) %>%\n",
    "                   arrange(position) %>%\n",
    "                   data.frame()\n",
    "  \n",
    "    xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID, \n",
    "                          beta = BETA,\n",
    "                          varbeta = (BETA_SE**2)) %>%\n",
    "                   select(rsID, snp, position, beta, varbeta,P)\n",
    "    \n",
    "    # check dimensions\n",
    "    dim(eqtl_subset)\n",
    "    dim(xwas_subset)\n",
    "        \n",
    "    # convert to acceptable data structure for input in coloc\n",
    "    eqtl_subset_list <- as.list(eqtl_subset)\n",
    "    eqtl_subset_list$type <- \"quant\"\n",
    "    xwas_subset_list <- as.list(xwas_subset)\n",
    "    xwas_subset_list$type <- \"cc\"\n",
    "    xwas_subset_list$N <- 2591 + 4023\n",
    "    #check_dataset(eqtl_subset_list)\n",
    "    #check_dataset(xwas_subset_list)\n",
    "        \n",
    "    # run coloc.abf\n",
    "    print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
    "    my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
    "                    dataset2=eqtl_subset_list)\n",
    "    print(my.res)\n",
    "    output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
    "\n",
    "    summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
    "    priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    \n",
    "    write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
    "}\n",
    "\n",
    "for (i in 1:length(brain_tissues)){\n",
    "    run_coloc(\"rs141773145\", i)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### index_var2 (rs6648060)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[-] Unloading gcc  11.3.0  ... \n",
      "[-] Unloading HDF5  1.12.2 \n",
      "[-] Unloading netcdf  4.9.0 \n",
      "[-] Unloading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[-] Unloading pandoc  2.18  on cn3180 \n",
      "[-] Unloading R 4.3.0 \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading HDF5  1.12.2 \n",
      "[+] Loading netcdf  4.9.0 \n",
      "[-] Unloading gcc  11.3.0  ... \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[+] Loading pandoc  2.18  on cn3180 \n",
      "[+] Loading R 4.3.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.3.0 (2023-04-21) -- \"Already Tomorrow\"\n",
      "Copyright (C) 2023 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.4\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.4.4     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.3     ✔ tidyr     1.3.0\n",
      "✔ purrr     1.0.2     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::between()     masks data.table::between()\n",
      "✖ dplyr::filter()      masks stats::filter()\n",
      "✖ dplyr::first()       masks data.table::first()\n",
      "✖ lubridate::hour()    masks data.table::hour()\n",
      "✖ lubridate::isoweek() masks data.table::isoweek()\n",
      "✖ dplyr::lag()         masks stats::lag()\n",
      "✖ dplyr::last()        masks data.table::last()\n",
      "✖ lubridate::mday()    masks data.table::mday()\n",
      "✖ lubridate::minute()  masks data.table::minute()\n",
      "✖ lubridate::month()   masks data.table::month()\n",
      "✖ lubridate::quarter() masks data.table::quarter()\n",
      "✖ lubridate::second()  masks data.table::second()\n",
      "✖ purrr::transpose()   masks data.table::transpose()\n",
      "✖ lubridate::wday()    masks data.table::wday()\n",
      "✖ lubridate::week()    masks data.table::week()\n",
      "✖ lubridate::yday()    masks data.table::yday()\n",
      "✖ lubridate::year()    masks data.table::year()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> # install.packages(\"coloc\")\n",
      "> require(coloc)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: coloc\n",
      "This is coloc version 5.2.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
      "> require(ieugwasr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ieugwasr\n",
      "API: public: http://gwas-api.mrcieu.ac.uk/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
      "> genetics.binaRies::get_plink_binary()\n",
      "[1] \"/gpfs/gsfs9/users/chiarp/R/rhel8/4.3/genetics.binaRies/bin/plink\"\n",
      "> \n",
      "> brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
      "> \n",
      "> run_coloc <- function(rsid,i){\n",
      "+     # read in files\n",
      "+     tissue <- brain_tissues[i]\n",
      "+     xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     dim(eqtl)\n",
      "+     dim(xwas)    \n",
      "+     \n",
      "+     # prep so that only columns that are needed for coloc is present\n",
      "+     eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID) %>%\n",
      "+                    filter(!is.na(beta)) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
      "+                    arrange(snp,pval_nominal) %>%\n",
      "+                    group_by(snp) %>%\n",
      "+                    slice(1:1) %>%\n",
      "+                    arrange(position) %>%\n",
      "+                    data.frame()\n",
      "+   \n",
      "+     xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID, \n",
      "+                           beta = BETA,\n",
      "+                           varbeta = (BETA_SE**2)) %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta,P)\n",
      "+     \n",
      "+     # check dimensions\n",
      "+     dim(eqtl_subset)\n",
      "+     dim(xwas_subset)\n",
      "+         \n",
      "+     # convert to acceptable data structure for input in coloc\n",
      "+     eqtl_subset_list <- as.list(eqtl_subset)\n",
      "+     eqtl_subset_list$type <- \"quant\"\n",
      "+     xwas_subset_list <- as.list(xwas_subset)\n",
      "+     xwas_subset_list$type <- \"cc\"\n",
      "+     xwas_subset_list$N <- 2591 + 4023\n",
      "+     #check_dataset(eqtl_subset_list)\n",
      "+     #check_dataset(xwas_subset_list)\n",
      "+         \n",
      "+     # run coloc.abf\n",
      "+     print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
      "+     my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
      "+                     dataset2=eqtl_subset_list)\n",
      "+     print(my.res)\n",
      "+     output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
      "+ \n",
      "+     summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
      "+     priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     \n",
      "+     write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
      "+ }\n",
      "> \n",
      "> for (i in 1:length(brain_tissues)){\n",
      "+     run_coloc(\"rs6648060\", i)\n",
      "+ }\n",
      "[1] \"run coloc for: Brain_Amygdala\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1900    0.3010    0.1810    0.2870    0.0413 \n",
      "[1] \"PP abf for shared variant: 4.13%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.07396756 0.4253 0.4253 0.1808376 0.04253\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4253.0000000    0.1895181    0.3011238    0.1807952    0.2872227    0.0413402 \n",
      "[1] \"saving coloc results for: Brain_Amygdala\"\n",
      "[1] \"run coloc for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "    0.106     0.167     0.269     0.425     0.033 \n",
      "[1] \"PP abf for shared variant: 3.3%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.065714 0.4225 0.4225 0.178464 0.04225\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.225000e+03 1.055683e-01 1.670299e-01 2.689456e-01 4.254920e-01 3.296408e-02 \n",
      "[1] \"saving coloc results for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "[1] \"run coloc for: Brain_Caudate_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1130    0.1800    0.2580    0.4100    0.0393 \n",
      "[1] \"PP abf for shared variant: 3.93%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.08135006 0.4278 0.4278 0.1829701 0.04278\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4278.0000000    0.1131746    0.1801491    0.2575021    0.4098477    0.0393265 \n",
      "[1] \"saving coloc results for: Brain_Caudate_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Cerebellar_Hemisphere\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0251    0.0400    0.3500    0.5570    0.0276 \n",
      "[1] \"PP abf for shared variant: 2.76%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.08135006 0.4278 0.4278 0.1829701 0.04278\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.278000e+03 2.513740e-02 4.001470e-02 3.500654e-01 5.572202e-01 2.756229e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellar_Hemisphere\"\n",
      "[1] \"run coloc for: Brain_Cerebellum\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.00127   0.00203   0.36800   0.58600   0.04300 \n",
      "[1] \"PP abf for shared variant: 4.3%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.07987256 0.4273 0.4273 0.1825426 0.04273\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.273000e+03 1.273431e-03 2.026589e-03 3.680514e-01 5.856890e-01 4.295959e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellum\"\n",
      "[1] \"run coloc for: Brain_Cortex\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0901    0.1430    0.2780    0.4430    0.0456 \n",
      "[1] \"PP abf for shared variant: 4.56%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.0822368 0.4281 0.4281 0.1832268 0.04281\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.281000e+03 9.006886e-02 1.433961e-01 2.781524e-01 4.427930e-01 4.558967e-02 \n",
      "[1] \"saving coloc results for: Brain_Cortex\"\n",
      "[1] \"run coloc for: Brain_Frontal_Cortex_BA9\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0182    0.0289    0.3570    0.5680    0.0283 \n",
      "[1] \"PP abf for shared variant: 2.83%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.07514792 0.4257 0.4257 0.1811779 0.04257\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.257000e+03 1.820608e-02 2.894195e-02 3.570127e-01 5.675097e-01 2.832954e-02 \n",
      "[1] \"saving coloc results for: Brain_Frontal_Cortex_BA9\"\n",
      "[1] \"run coloc for: Brain_Hippocampus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.2350    0.3730    0.1350    0.2140    0.0434 \n",
      "[1] \"PP abf for shared variant: 4.34%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.0763286 0.4261 0.4261 0.1815186 0.04261\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.261000e+03 2.346226e-01 3.729331e-01 1.347963e-01 2.142155e-01 4.343244e-02 \n",
      "[1] \"saving coloc results for: Brain_Hippocampus\"\n",
      "[1] \"run coloc for: Brain_Hypothalamus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1520    0.2410    0.2150    0.3420    0.0499 \n",
      "[1] \"PP abf for shared variant: 4.99%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.0704284 0.4241 0.4241 0.1798184 0.04241\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.241000e+03 1.519727e-01 2.413017e-01 2.151960e-01 3.416375e-01 4.989211e-02 \n",
      "[1] \"saving coloc results for: Brain_Hypothalamus\"\n",
      "[1] \"run coloc for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0569    0.0904    0.3160    0.5020    0.0350 \n",
      "[1] \"PP abf for shared variant: 3.5%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.07721432 0.4264 0.4264 0.1817743 0.04264\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.264000e+03 5.685320e-02 9.043541e-02 3.156610e-01 5.020816e-01 3.496874e-02 \n",
      "[1] \"saving coloc results for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Putamen_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0632    0.1000    0.3120    0.4950    0.0298 \n",
      "[1] \"PP abf for shared variant: 2.98%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0    H1    H2        H3     H4\n",
      " -0.0730825 0.425 0.425 0.1805825 0.0425\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.250000e+03 6.321164e-02 1.004213e-01 3.115893e-01 4.949770e-01 2.980087e-02 \n",
      "[1] \"saving coloc results for: Brain_Putamen_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Spinal_cord_cervical_c-1\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0147    0.0234    0.3620    0.5750    0.0249 \n",
      "[1] \"PP abf for shared variant: 2.49%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.0745577 0.4255 0.4255 0.1810077 0.04255\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.255000e+03 1.474501e-02 2.342721e-02 3.619185e-01 5.749994e-01 2.490988e-02 \n",
      "[1] \"saving coloc results for: Brain_Spinal_cord_cervical_c-1\"\n",
      "[1] \"run coloc for: Brain_Substantia_nigra\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1660    0.2640    0.2020    0.3210    0.0482 \n",
      "[1] \"PP abf for shared variant: 4.82%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "          H0     H1     H2        H3      H4\n",
      " -0.07278752 0.4249 0.4249 0.1804975 0.04249\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.249000e+03 1.659074e-01 2.635033e-01 2.018544e-01 3.205481e-01 4.818679e-02 \n",
      "[1] \"saving coloc results for: Brain_Substantia_nigra\"\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 29 warnings (use warnings() to see them)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/females_only_cond_ApoE4/\n",
    "\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "# install.packages(\"coloc\")\n",
    "require(coloc)\n",
    "#devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
    "require(ieugwasr)\n",
    "#devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
    "genetics.binaRies::get_plink_binary()\n",
    "\n",
    "brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
    "\n",
    "run_coloc <- function(rsid,i){\n",
    "    # read in files\n",
    "    tissue <- brain_tissues[i]\n",
    "    xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    dim(eqtl)\n",
    "    dim(xwas)    \n",
    "    \n",
    "    # prep so that only columns that are needed for coloc is present\n",
    "    eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID) %>%\n",
    "                   filter(!is.na(beta)) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
    "                   arrange(snp,pval_nominal) %>%\n",
    "                   group_by(snp) %>%\n",
    "                   slice(1:1) %>%\n",
    "                   arrange(position) %>%\n",
    "                   data.frame()\n",
    "  \n",
    "    xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID, \n",
    "                          beta = BETA,\n",
    "                          varbeta = (BETA_SE**2)) %>%\n",
    "                   select(rsID, snp, position, beta, varbeta,P)\n",
    "    \n",
    "    # check dimensions\n",
    "    dim(eqtl_subset)\n",
    "    dim(xwas_subset)\n",
    "        \n",
    "    # convert to acceptable data structure for input in coloc\n",
    "    eqtl_subset_list <- as.list(eqtl_subset)\n",
    "    eqtl_subset_list$type <- \"quant\"\n",
    "    xwas_subset_list <- as.list(xwas_subset)\n",
    "    xwas_subset_list$type <- \"cc\"\n",
    "    xwas_subset_list$N <- 2591 + 4023\n",
    "    #check_dataset(eqtl_subset_list)\n",
    "    #check_dataset(xwas_subset_list)\n",
    "        \n",
    "    # run coloc.abf\n",
    "    print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
    "    my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
    "                    dataset2=eqtl_subset_list)\n",
    "    print(my.res)\n",
    "    output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
    "\n",
    "    summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
    "    priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    \n",
    "    write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
    "}\n",
    "\n",
    "for (i in 1:length(brain_tissues)){\n",
    "    run_coloc(\"rs6648060\", i)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### index_var3 (rs141193614)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[-] Unloading gcc  11.3.0  ... \n",
      "[-] Unloading HDF5  1.12.2 \n",
      "[-] Unloading netcdf  4.9.0 \n",
      "[-] Unloading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[-] Unloading pandoc  2.18  on cn3180 \n",
      "[-] Unloading R 4.3.0 \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading HDF5  1.12.2 \n",
      "[+] Loading netcdf  4.9.0 \n",
      "[-] Unloading gcc  11.3.0  ... \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[+] Loading pandoc  2.18  on cn3180 \n",
      "[+] Loading R 4.3.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.3.0 (2023-04-21) -- \"Already Tomorrow\"\n",
      "Copyright (C) 2023 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.4\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.4.4     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.3     ✔ tidyr     1.3.0\n",
      "✔ purrr     1.0.2     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::between()     masks data.table::between()\n",
      "✖ dplyr::filter()      masks stats::filter()\n",
      "✖ dplyr::first()       masks data.table::first()\n",
      "✖ lubridate::hour()    masks data.table::hour()\n",
      "✖ lubridate::isoweek() masks data.table::isoweek()\n",
      "✖ dplyr::lag()         masks stats::lag()\n",
      "✖ dplyr::last()        masks data.table::last()\n",
      "✖ lubridate::mday()    masks data.table::mday()\n",
      "✖ lubridate::minute()  masks data.table::minute()\n",
      "✖ lubridate::month()   masks data.table::month()\n",
      "✖ lubridate::quarter() masks data.table::quarter()\n",
      "✖ lubridate::second()  masks data.table::second()\n",
      "✖ purrr::transpose()   masks data.table::transpose()\n",
      "✖ lubridate::wday()    masks data.table::wday()\n",
      "✖ lubridate::week()    masks data.table::week()\n",
      "✖ lubridate::yday()    masks data.table::yday()\n",
      "✖ lubridate::year()    masks data.table::year()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> # install.packages(\"coloc\")\n",
      "> require(coloc)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: coloc\n",
      "This is coloc version 5.2.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
      "> require(ieugwasr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ieugwasr\n",
      "API: public: http://gwas-api.mrcieu.ac.uk/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
      "> genetics.binaRies::get_plink_binary()\n",
      "[1] \"/gpfs/gsfs9/users/chiarp/R/rhel8/4.3/genetics.binaRies/bin/plink\"\n",
      "> \n",
      "> brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
      "> \n",
      "> run_coloc <- function(rsid,i){\n",
      "+     # read in files\n",
      "+     tissue <- brain_tissues[i]\n",
      "+     xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     dim(eqtl)\n",
      "+     dim(xwas)    \n",
      "+     \n",
      "+     # prep so that only columns that are needed for coloc is present\n",
      "+     eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID) %>%\n",
      "+                    filter(!is.na(beta)) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
      "+                    arrange(snp,pval_nominal) %>%\n",
      "+                    group_by(snp) %>%\n",
      "+                    slice(1:1) %>%\n",
      "+                    arrange(position) %>%\n",
      "+                    data.frame()\n",
      "+   \n",
      "+     xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID, \n",
      "+                           beta = BETA,\n",
      "+                           varbeta = (BETA_SE**2)) %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta,P)\n",
      "+     \n",
      "+     # check dimensions\n",
      "+     dim(eqtl_subset)\n",
      "+     dim(xwas_subset)\n",
      "+         \n",
      "+     # convert to acceptable data structure for input in coloc\n",
      "+     eqtl_subset_list <- as.list(eqtl_subset)\n",
      "+     eqtl_subset_list$type <- \"quant\"\n",
      "+     xwas_subset_list <- as.list(xwas_subset)\n",
      "+     xwas_subset_list$type <- \"cc\"\n",
      "+     xwas_subset_list$N <- 2591 + 4023\n",
      "+     #check_dataset(eqtl_subset_list)\n",
      "+     #check_dataset(xwas_subset_list)\n",
      "+         \n",
      "+     # run coloc.abf\n",
      "+     print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
      "+     my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
      "+                     dataset2=eqtl_subset_list)\n",
      "+     print(my.res)\n",
      "+     output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
      "+ \n",
      "+     summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
      "+     priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     \n",
      "+     write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
      "+ }\n",
      "> \n",
      "> for (i in 1:length(brain_tissues)){\n",
      "+     run_coloc(\"rs141193614\", i)\n",
      "+ }\n",
      "[1] \"run coloc for: Brain_Amygdala\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.89700   0.04410   0.05170   0.00254   0.00419 \n",
      "[1] \"PP abf for shared variant: 0.419%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "       H0     H1     H2         H3      H4\n",
      " 0.869962 0.0602 0.0602 0.00361802 0.00602\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "6.020000e+02 8.974496e-01 4.408782e-02 5.173372e-02 2.537262e-03 4.191582e-03 \n",
      "[1] \"saving coloc results for: Brain_Amygdala\"\n",
      "[1] \"run coloc for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 0.953000  0.022700  0.021400  0.000509  0.002130 \n",
      "[1] \"PP abf for shared variant: 0.213%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.9520315 0.0226 0.0226 0.0005085 0.00226\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.260000e+02 9.532109e-01 2.272881e-02 2.142625e-02 5.087724e-04 2.125248e-03 \n",
      "[1] \"saving coloc results for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "[1] \"run coloc for: Brain_Caudate_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.89200   0.04420   0.05620   0.00278   0.00448 \n",
      "[1] \"PP abf for shared variant: 0.448%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.8688516 0.0607 0.0607 0.00367842 0.00607\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "6.070000e+02 8.923663e-01 4.420335e-02 5.617161e-02 2.777979e-03 4.480789e-03 \n",
      "[1] \"saving coloc results for: Brain_Caudate_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Cerebellar_Hemisphere\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 0.954000  0.022800  0.020400  0.000484  0.002120 \n",
      "[1] \"PP abf for shared variant: 0.212%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.9520315 0.0226 0.0226 0.0005085 0.00226\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.260000e+02 9.542574e-01 2.275376e-02 2.038152e-02 4.838630e-04 2.123422e-03 \n",
      "[1] \"saving coloc results for: Brain_Cerebellar_Hemisphere\"\n",
      "[1] \"run coloc for: Brain_Cerebellum\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 0.953000  0.022800  0.021000  0.000501  0.002250 \n",
      "[1] \"PP abf for shared variant: 0.225%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "       H0     H1     H2         H3      H4\n",
      " 0.951817 0.0227 0.0227 0.00051302 0.00227\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.270000e+02 9.533992e-01 2.283337e-02 2.101768e-02 5.011130e-04 2.248664e-03 \n",
      "[1] \"saving coloc results for: Brain_Cerebellum\"\n",
      "[1] \"run coloc for: Brain_Cortex\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 0.953000  0.022800  0.021400  0.000509  0.002460 \n",
      "[1] \"PP abf for shared variant: 0.246%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "       H0     H1     H2         H3      H4\n",
      " 0.951817 0.0227 0.0227 0.00051302 0.00227\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.270000e+02 9.528488e-01 2.282019e-02 2.136247e-02 5.091597e-04 2.459371e-03 \n",
      "[1] \"saving coloc results for: Brain_Cortex\"\n",
      "[1] \"run coloc for: Brain_Frontal_Cortex_BA9\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 0.954000  0.022800  0.020400  0.000487  0.002140 \n",
      "[1] \"PP abf for shared variant: 0.214%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "       H0     H1     H2         H3      H4\n",
      " 0.951817 0.0227 0.0227 0.00051302 0.00227\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.270000e+02 9.540919e-01 2.284996e-02 2.042826e-02 4.871025e-04 2.142781e-03 \n",
      "[1] \"saving coloc results for: Brain_Frontal_Cortex_BA9\"\n",
      "[1] \"run coloc for: Brain_Hippocampus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.88200   0.04360   0.06610   0.00326   0.00498 \n",
      "[1] \"PP abf for shared variant: 0.498%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.8690737 0.0606 0.0606 0.0036663 0.00606\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "6.060000e+02 8.820779e-01 4.362620e-02 6.605071e-02 3.261782e-03 4.983382e-03 \n",
      "[1] \"saving coloc results for: Brain_Hippocampus\"\n",
      "[1] \"run coloc for: Brain_Hypothalamus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.89000   0.04400   0.05810   0.00287   0.00452 \n",
      "[1] \"PP abf for shared variant: 0.452%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.8690737 0.0606 0.0606 0.0036663 0.00606\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "6.060000e+02 8.904821e-01 4.404311e-02 5.808672e-02 2.868441e-03 4.519617e-03 \n",
      "[1] \"saving coloc results for: Brain_Hypothalamus\"\n",
      "[1] \"run coloc for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.88700   0.04400   0.06140   0.00304   0.00456 \n",
      "[1] \"PP abf for shared variant: 0.456%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.8686294 0.0608 0.0608 0.00369056 0.00608\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "6.080000e+02 8.869495e-01 4.402813e-02 6.141559e-02 3.044104e-03 4.562627e-03 \n",
      "[1] \"saving coloc results for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Putamen_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.89400   0.04410   0.05490   0.00271   0.00451 \n",
      "[1] \"PP abf for shared variant: 0.451%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2        H3      H4\n",
      " 0.8690737 0.0606 0.0606 0.0036663 0.00606\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "6.060000e+02 8.937788e-01 4.413670e-02 5.487266e-02 2.705222e-03 4.506571e-03 \n",
      "[1] \"saving coloc results for: Brain_Putamen_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Spinal_cord_cervical_c-1\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "  0.80400   0.07470   0.10400   0.00970   0.00726 \n",
      "[1] \"PP abf for shared variant: 0.726%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2         H3      H4\n",
      " 0.6914074 0.1379 0.1379 0.01900262 0.01379\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "1.379000e+03 8.039040e-01 7.466283e-02 1.044755e-01 9.695929e-03 7.261761e-03 \n",
      "[1] \"saving coloc results for: Brain_Spinal_cord_cervical_c-1\"\n",
      "[1] \"run coloc for: Brain_Substantia_nigra\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      " 0.956000  0.022200  0.019700  0.000455  0.002100 \n",
      "[1] \"PP abf for shared variant: 0.21%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0    H1    H2        H3     H4\n",
      " 0.9533182 0.022 0.022 0.0004818 0.0022\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "2.200000e+02 9.555769e-01 2.217815e-02 1.969396e-02 4.549845e-04 2.095985e-03 \n",
      "[1] \"saving coloc results for: Brain_Substantia_nigra\"\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 39 warnings (use warnings() to see them)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/females_only_cond_ApoE4/\n",
    "\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "# install.packages(\"coloc\")\n",
    "require(coloc)\n",
    "#devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
    "require(ieugwasr)\n",
    "#devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
    "genetics.binaRies::get_plink_binary()\n",
    "\n",
    "brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
    "\n",
    "run_coloc <- function(rsid,i){\n",
    "    # read in files\n",
    "    tissue <- brain_tissues[i]\n",
    "    xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    dim(eqtl)\n",
    "    dim(xwas)    \n",
    "    \n",
    "    # prep so that only columns that are needed for coloc is present\n",
    "    eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID) %>%\n",
    "                   filter(!is.na(beta)) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
    "                   arrange(snp,pval_nominal) %>%\n",
    "                   group_by(snp) %>%\n",
    "                   slice(1:1) %>%\n",
    "                   arrange(position) %>%\n",
    "                   data.frame()\n",
    "  \n",
    "    xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID, \n",
    "                          beta = BETA,\n",
    "                          varbeta = (BETA_SE**2)) %>%\n",
    "                   select(rsID, snp, position, beta, varbeta,P)\n",
    "    \n",
    "    # check dimensions\n",
    "    dim(eqtl_subset)\n",
    "    dim(xwas_subset)\n",
    "        \n",
    "    # convert to acceptable data structure for input in coloc\n",
    "    eqtl_subset_list <- as.list(eqtl_subset)\n",
    "    eqtl_subset_list$type <- \"quant\"\n",
    "    xwas_subset_list <- as.list(xwas_subset)\n",
    "    xwas_subset_list$type <- \"cc\"\n",
    "    xwas_subset_list$N <- 2591 + 4023\n",
    "    #check_dataset(eqtl_subset_list)\n",
    "    #check_dataset(xwas_subset_list)\n",
    "        \n",
    "    # run coloc.abf\n",
    "    print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
    "    my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
    "                    dataset2=eqtl_subset_list)\n",
    "    print(my.res)\n",
    "    output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
    "\n",
    "    summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
    "    priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    \n",
    "    write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
    "}\n",
    "\n",
    "for (i in 1:length(brain_tissues)){\n",
    "    run_coloc(\"rs141193614\", i)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### index_var4 (rs12860838)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[-] Unloading gcc  11.3.0  ... \n",
      "[-] Unloading HDF5  1.12.2 \n",
      "[-] Unloading netcdf  4.9.0 \n",
      "[-] Unloading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[-] Unloading pandoc  2.18  on cn3180 \n",
      "[-] Unloading R 4.3.0 \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading HDF5  1.12.2 \n",
      "[+] Loading netcdf  4.9.0 \n",
      "[-] Unloading gcc  11.3.0  ... \n",
      "[+] Loading gcc  11.3.0  ... \n",
      "[+] Loading openmpi/4.1.3/gcc-11.3.0  ... \n",
      "[+] Loading pandoc  2.18  on cn3180 \n",
      "[+] Loading R 4.3.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.3.0 (2023-04-21) -- \"Already Tomorrow\"\n",
      "Copyright (C) 2023 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.4\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.4.4     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.3     ✔ tidyr     1.3.0\n",
      "✔ purrr     1.0.2     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::between()     masks data.table::between()\n",
      "✖ dplyr::filter()      masks stats::filter()\n",
      "✖ dplyr::first()       masks data.table::first()\n",
      "✖ lubridate::hour()    masks data.table::hour()\n",
      "✖ lubridate::isoweek() masks data.table::isoweek()\n",
      "✖ dplyr::lag()         masks stats::lag()\n",
      "✖ dplyr::last()        masks data.table::last()\n",
      "✖ lubridate::mday()    masks data.table::mday()\n",
      "✖ lubridate::minute()  masks data.table::minute()\n",
      "✖ lubridate::month()   masks data.table::month()\n",
      "✖ lubridate::quarter() masks data.table::quarter()\n",
      "✖ lubridate::second()  masks data.table::second()\n",
      "✖ purrr::transpose()   masks data.table::transpose()\n",
      "✖ lubridate::wday()    masks data.table::wday()\n",
      "✖ lubridate::week()    masks data.table::week()\n",
      "✖ lubridate::yday()    masks data.table::yday()\n",
      "✖ lubridate::year()    masks data.table::year()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> # install.packages(\"coloc\")\n",
      "> require(coloc)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: coloc\n",
      "This is coloc version 5.2.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
      "> require(ieugwasr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ieugwasr\n",
      "API: public: http://gwas-api.mrcieu.ac.uk/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> #devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
      "> genetics.binaRies::get_plink_binary()\n",
      "[1] \"/gpfs/gsfs9/users/chiarp/R/rhel8/4.3/genetics.binaRies/bin/plink\"\n",
      "> \n",
      "> brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
      "> \n",
      "> run_coloc <- function(rsid,i){\n",
      "+     # read in files\n",
      "+     tissue <- brain_tissues[i]\n",
      "+     xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
      "+     dim(eqtl)\n",
      "+     dim(xwas)    \n",
      "+     \n",
      "+     # prep so that only columns that are needed for coloc is present\n",
      "+     eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID) %>%\n",
      "+                    filter(!is.na(beta)) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
      "+                    arrange(snp,pval_nominal) %>%\n",
      "+                    group_by(snp) %>%\n",
      "+                    slice(1:1) %>%\n",
      "+                    arrange(position) %>%\n",
      "+                    data.frame()\n",
      "+   \n",
      "+     xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
      "+                    filter(rsID != \"\") %>%\n",
      "+                    mutate(position = POS,\n",
      "+                           snp = ID, \n",
      "+                           beta = BETA,\n",
      "+                           varbeta = (BETA_SE**2)) %>%\n",
      "+                    select(rsID, snp, position, beta, varbeta,P)\n",
      "+     \n",
      "+     # check dimensions\n",
      "+     dim(eqtl_subset)\n",
      "+     dim(xwas_subset)\n",
      "+         \n",
      "+     # convert to acceptable data structure for input in coloc\n",
      "+     eqtl_subset_list <- as.list(eqtl_subset)\n",
      "+     eqtl_subset_list$type <- \"quant\"\n",
      "+     xwas_subset_list <- as.list(xwas_subset)\n",
      "+     xwas_subset_list$type <- \"cc\"\n",
      "+     xwas_subset_list$N <- 2591 + 4023\n",
      "+     #check_dataset(eqtl_subset_list)\n",
      "+     #check_dataset(xwas_subset_list)\n",
      "+         \n",
      "+     # run coloc.abf\n",
      "+     print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
      "+     my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
      "+                     dataset2=eqtl_subset_list)\n",
      "+     print(my.res)\n",
      "+     output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
      "+ \n",
      "+     summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
      "+     priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
      "+     \n",
      "+     write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
      "+     print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
      "+ }\n",
      "> \n",
      "> for (i in 1:length(brain_tissues)){\n",
      "+     run_coloc(\"rs12860838\", i)\n",
      "+ }\n",
      "[1] \"run coloc for: Brain_Amygdala\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1630    0.4220    0.1040    0.2680    0.0428 \n",
      "[1] \"PP abf for shared variant: 4.28%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2558259 0.4857 0.4857 0.2358559 0.04857\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.857000e+03 1.631117e-01 4.220668e-01 1.036972e-01 2.682835e-01 4.284086e-02 \n",
      "[1] \"saving coloc results for: Brain_Amygdala\"\n",
      "[1] \"run coloc for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1460    0.3770    0.1220    0.3160    0.0398 \n",
      "[1] \"PP abf for shared variant: 3.98%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0    H1    H2        H3     H4\n",
      " -0.2659721 0.489 0.489 0.2390721 0.0489\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.890000e+03 1.455758e-01 3.771412e-01 1.218449e-01 3.156222e-01 3.981586e-02 \n",
      "[1] \"saving coloc results for: Brain_Anterior_cingulate_cortex_BA24\"\n",
      "[1] \"run coloc for: Brain_Caudate_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1510    0.3920    0.1130    0.2920    0.0517 \n",
      "[1] \"PP abf for shared variant: 5.17%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2684351 0.4898 0.4898 0.2398551 0.04898\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.898000e+03 1.514486e-01 3.924806e-01 1.126149e-01 2.917911e-01 5.166476e-02 \n",
      "[1] \"saving coloc results for: Brain_Caudate_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Cerebellar_Hemisphere\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.0166    0.0430    0.2470    0.6390    0.0547 \n",
      "[1] \"PP abf for shared variant: 5.47%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2675113 0.4895 0.4895 0.2395613 0.04895\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.895000e+03 1.659034e-02 4.299379e-02 2.466217e-01 6.390644e-01 5.472982e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellar_Hemisphere\"\n",
      "[1] \"run coloc for: Brain_Cerebellum\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1030    0.2680    0.1590    0.4120    0.0579 \n",
      "[1] \"PP abf for shared variant: 5.79%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.269359 0.4901 0.4901 0.240149 0.04901\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.901000e+03 1.034065e-01 2.680317e-01 1.588840e-01 4.117724e-01 5.790539e-02 \n",
      "[1] \"saving coloc results for: Brain_Cerebellum\"\n",
      "[1] \"run coloc for: Brain_Cortex\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "    0.128     0.333     0.115     0.299     0.125 \n",
      "[1] \"PP abf for shared variant: 12.5%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2678192 0.4896 0.4896 0.2396592 0.04896\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4896.0000000    0.1283041    0.3325170    0.1154377    0.2990473    0.1246939 \n",
      "[1] \"saving coloc results for: Brain_Cortex\"\n",
      "[1] \"run coloc for: Brain_Frontal_Cortex_BA9\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1610    0.4170    0.1040    0.2710    0.0473 \n",
      "[1] \"PP abf for shared variant: 4.73%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0   H1   H2       H3    H4\n",
      " -0.269051 0.49 0.49 0.240051 0.049\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.900000e+03 1.607783e-01 4.167250e-01 1.044717e-01 2.707354e-01 4.728951e-02 \n",
      "[1] \"saving coloc results for: Brain_Frontal_Cortex_BA9\"\n",
      "[1] \"run coloc for: Brain_Hippocampus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1350    0.3490    0.1330    0.3440    0.0397 \n",
      "[1] \"PP abf for shared variant: 3.97%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.269359 0.4901 0.4901 0.240149 0.04901\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.901000e+03 1.347504e-01 3.492862e-01 1.326013e-01 3.436759e-01 3.968615e-02 \n",
      "[1] \"saving coloc results for: Brain_Hippocampus\"\n",
      "[1] \"run coloc for: Brain_Hypothalamus\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1620    0.4190    0.1030    0.2670    0.0493 \n",
      "[1] \"PP abf for shared variant: 4.93%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.2665877 0.4892 0.4892 0.2392677 0.04892\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.892000e+03 1.615881e-01 4.186825e-01 1.031654e-01 2.672573e-01 4.930665e-02 \n",
      "[1] \"saving coloc results for: Brain_Hypothalamus\"\n",
      "[1] \"run coloc for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "    0.132     0.343     0.130     0.337     0.058 \n",
      "[1] \"PP abf for shared variant: 5.8%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "        H0     H1     H2       H3      H4\n",
      " -0.269667 0.4902 0.4902 0.240247 0.04902\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.902000e+03 1.323240e-01 3.430071e-01 1.299352e-01 3.367570e-01 5.797667e-02 \n",
      "[1] \"saving coloc results for: Brain_Nucleus_accumbens_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Putamen_basal_ganglia\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1450    0.3750    0.1170    0.3030    0.0607 \n",
      "[1] \"PP abf for shared variant: 6.07%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0    H1    H2        H3     H4\n",
      " -0.2659721 0.489 0.489 0.2390721 0.0489\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.890000e+03 1.445514e-01 3.745368e-01 1.170300e-01 3.031674e-01 6.071437e-02 \n",
      "[1] \"saving coloc results for: Brain_Putamen_basal_ganglia\"\n",
      "[1] \"run coloc for: Brain_Spinal_cord_cervical_c-1\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1600    0.4130    0.1090    0.2810    0.0382 \n",
      "[1] \"PP abf for shared variant: 3.82%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0    H1    H2        H3     H4\n",
      " -0.2567474 0.486 0.486 0.2361474 0.0486\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.860000e+03 1.595135e-01 4.127673e-01 1.085782e-01 2.809256e-01 3.821548e-02 \n",
      "[1] \"saving coloc results for: Brain_Spinal_cord_cervical_c-1\"\n",
      "[1] \"run coloc for: Brain_Substantia_nigra\"\n",
      "PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf \n",
      "   0.1710    0.4370    0.0911    0.2330    0.0676 \n",
      "[1] \"PP abf for shared variant: 6.76%\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coloc analysis of trait 1, trait 2\n",
      "\n",
      "SNP Priors\n",
      "\n",
      "Hypothesis Priors\n",
      "\n",
      "Posterior\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p1    p2   p12 \n",
      "1e-04 1e-04 1e-05 \n",
      "         H0     H1     H2        H3      H4\n",
      " -0.1133861 0.4386 0.4386 0.1923261 0.04386\n",
      "       nsnps           H0           H1           H2           H3           H4 \n",
      "4.386000e+03 1.707745e-01 4.374195e-01 9.105843e-02 2.331683e-01 6.757928e-02 \n",
      "[1] \"saving coloc results for: Brain_Substantia_nigra\"\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 38 warnings (use warnings() to see them)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/females_only_cond_ApoE4/\n",
    "\n",
    "module load R/4.3\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "# install.packages(\"coloc\")\n",
    "require(coloc)\n",
    "#devtools::install_github(\"mrcieu/ieugwasr\", force=TRUE)\n",
    "require(ieugwasr)\n",
    "#devtools::install_github(\"explodecomputer/genetics.binaRies\")\n",
    "genetics.binaRies::get_plink_binary()\n",
    "\n",
    "brain_tissues <- c(\"Brain_Amygdala\",\"Brain_Anterior_cingulate_cortex_BA24\",\"Brain_Caudate_basal_ganglia\",\"Brain_Cerebellar_Hemisphere\",\"Brain_Cerebellum\",\"Brain_Cortex\",\"Brain_Frontal_Cortex_BA9\",\"Brain_Hippocampus\",\"Brain_Hypothalamus\",\"Brain_Nucleus_accumbens_basal_ganglia\",\"Brain_Putamen_basal_ganglia\",\"Brain_Spinal_cord_cervical_c-1\",\"Brain_Substantia_nigra\")\n",
    "\n",
    "run_coloc <- function(rsid,i){\n",
    "    # read in files\n",
    "    tissue <- brain_tissues[i]\n",
    "    xwas <- fread(paste(\"XWAS_summ_stats_\",rsid,\"_flank1Mb_no-maf-filter.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    eqtl <- fread(paste(\"../GTEXv8_eQTL_brains/\",tissue,\".v8.EUR.allpairs.chrX_harmonized_snpid_shared.txt\",sep=\"\")) %>% arrange(POS)\n",
    "    dim(eqtl)\n",
    "    dim(xwas)    \n",
    "    \n",
    "    # prep so that only columns that are needed for coloc is present\n",
    "    eqtl_subset <- subset(eqtl, eqtl$ID %in% xwas$ID) %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID) %>%\n",
    "                   filter(!is.na(beta)) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   select(rsID, snp, position, beta, varbeta, N, MAF, phenotype_id, pval_nominal) %>%\n",
    "                   arrange(snp,pval_nominal) %>%\n",
    "                   group_by(snp) %>%\n",
    "                   slice(1:1) %>%\n",
    "                   arrange(position) %>%\n",
    "                   data.frame()\n",
    "  \n",
    "    xwas_subset <- subset(xwas, xwas$ID %in% eqtl_subset$snp) %>%\n",
    "                   filter(rsID != \"\") %>%\n",
    "                   mutate(position = POS,\n",
    "                          snp = ID, \n",
    "                          beta = BETA,\n",
    "                          varbeta = (BETA_SE**2)) %>%\n",
    "                   select(rsID, snp, position, beta, varbeta,P)\n",
    "    \n",
    "    # check dimensions\n",
    "    dim(eqtl_subset)\n",
    "    dim(xwas_subset)\n",
    "        \n",
    "    # convert to acceptable data structure for input in coloc\n",
    "    eqtl_subset_list <- as.list(eqtl_subset)\n",
    "    eqtl_subset_list$type <- \"quant\"\n",
    "    xwas_subset_list <- as.list(xwas_subset)\n",
    "    xwas_subset_list$type <- \"cc\"\n",
    "    xwas_subset_list$N <- 2591 + 4023\n",
    "    #check_dataset(eqtl_subset_list)\n",
    "    #check_dataset(xwas_subset_list)\n",
    "        \n",
    "    # run coloc.abf\n",
    "    print(paste(\"run coloc for: \",tissue,sep=\"\"))\n",
    "    my.res <- coloc.abf(dataset1=xwas_subset_list,\n",
    "                    dataset2=eqtl_subset_list)\n",
    "    print(my.res)\n",
    "    output_name <- paste(\"coloc_\",rsid,\"_\",tissue, sep=\"\")\n",
    "\n",
    "    summary <- my.res$summary %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    results <- my.res$results %>% data.frame() %>% mutate(Tissue = tissue,index_variant = rsid)\n",
    "    priors <- my.res$priors %>% data.frame() %>% t() %>% data.frame() %>% mutate(Tissue = tissue, index_variant = rsid)\n",
    "    \n",
    "    write.table(summary, paste(output_name,\"_summary.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(results, paste(output_name,\"_results.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    write.table(priors, paste(output_name,\"_priors.txt\",sep=\"\"), quote=F, sep=\"\\t\", row.names=F, col.names=T)\n",
    "    print(paste(\"saving coloc results for: \",tissue,sep=\"\"))\n",
    "}\n",
    "\n",
    "for (i in 1:length(brain_tissues)){\n",
    "    run_coloc(\"rs12860838\", i)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### merge coloc results and make posterior probability table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd Analysis.COLOC_redo/\n",
    "\n",
    "head -n 1 females_only/coloc_rs141773145_Brain_Amygdala_summary.txt > females_only/coloc_posteriors_compiled_allIndexVars_allBrainTissues.txt\n",
    "cat females_only/coloc_rs141773145_Brain_*_summary.txt females_only/coloc_rs12860838_Brain_*_summary.txt | grep -v \"Tissue\" >> females_only/coloc_posteriors_compiled_allIndexVars_allBrainTissues.txt\n",
    "\n",
    "\n",
    "head -n 1 females_only_cond_ApoE4/coloc_rs141773145_Brain_Amygdala_summary.txt > females_only_cond_ApoE4/coloc_posteriors_compiled_allIndexVars_allBrainTissues.txt\n",
    "cat females_only_cond_ApoE4/coloc_rs141773145_Brain_*_summary.txt females_only_cond_ApoE4/coloc_rs6648060_Brain_*_summary.txt females_only_cond_ApoE4/coloc_rs141193614_Brain_*_summary.txt females_only_cond_ApoE4/coloc_rs12860838_Brain_*_summary.txt | grep -v \"Tissue\" >> females_only_cond_ApoE4/coloc_posteriors_compiled_allIndexVars_allBrainTissues.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nsnps</th>\n",
       "      <th>PP.H0.abf</th>\n",
       "      <th>PP.H1.abf</th>\n",
       "      <th>PP.H2.abf</th>\n",
       "      <th>PP.H3.abf</th>\n",
       "      <th>PP.H4.abf</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>index_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2015</td>\n",
       "      <td>0.636188</td>\n",
       "      <td>0.098066</td>\n",
       "      <td>0.218374</td>\n",
       "      <td>0.033648</td>\n",
       "      <td>0.013724</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2022</td>\n",
       "      <td>0.644494</td>\n",
       "      <td>0.099806</td>\n",
       "      <td>0.209482</td>\n",
       "      <td>0.032427</td>\n",
       "      <td>0.013792</td>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2034</td>\n",
       "      <td>0.013330</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>0.825616</td>\n",
       "      <td>0.128778</td>\n",
       "      <td>0.030196</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2032</td>\n",
       "      <td>0.060078</td>\n",
       "      <td>0.009348</td>\n",
       "      <td>0.779921</td>\n",
       "      <td>0.121331</td>\n",
       "      <td>0.029322</td>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2035</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.839204</td>\n",
       "      <td>0.130955</td>\n",
       "      <td>0.029773</td>\n",
       "      <td>Brain_Cerebellum</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2035</td>\n",
       "      <td>0.459558</td>\n",
       "      <td>0.071636</td>\n",
       "      <td>0.389932</td>\n",
       "      <td>0.060765</td>\n",
       "      <td>0.018108</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2028</td>\n",
       "      <td>0.105517</td>\n",
       "      <td>0.016404</td>\n",
       "      <td>0.735825</td>\n",
       "      <td>0.114366</td>\n",
       "      <td>0.027888</td>\n",
       "      <td>Brain_Frontal_Cortex_BA9</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2035</td>\n",
       "      <td>0.469551</td>\n",
       "      <td>0.073282</td>\n",
       "      <td>0.379477</td>\n",
       "      <td>0.059206</td>\n",
       "      <td>0.018485</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2025</td>\n",
       "      <td>0.375991</td>\n",
       "      <td>0.058281</td>\n",
       "      <td>0.471092</td>\n",
       "      <td>0.073000</td>\n",
       "      <td>0.021637</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2035</td>\n",
       "      <td>0.183217</td>\n",
       "      <td>0.028599</td>\n",
       "      <td>0.660042</td>\n",
       "      <td>0.103003</td>\n",
       "      <td>0.025139</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2027</td>\n",
       "      <td>0.045461</td>\n",
       "      <td>0.007065</td>\n",
       "      <td>0.792213</td>\n",
       "      <td>0.123079</td>\n",
       "      <td>0.032183</td>\n",
       "      <td>Brain_Putamen_basal_ganglia</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2018</td>\n",
       "      <td>0.146741</td>\n",
       "      <td>0.022651</td>\n",
       "      <td>0.695379</td>\n",
       "      <td>0.107309</td>\n",
       "      <td>0.027920</td>\n",
       "      <td>Brain_Spinal_cord_cervical_c-1</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2006</td>\n",
       "      <td>0.653768</td>\n",
       "      <td>0.100076</td>\n",
       "      <td>0.201305</td>\n",
       "      <td>0.030801</td>\n",
       "      <td>0.014050</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4859</td>\n",
       "      <td>0.239418</td>\n",
       "      <td>0.348457</td>\n",
       "      <td>0.152259</td>\n",
       "      <td>0.221563</td>\n",
       "      <td>0.038303</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>4892</td>\n",
       "      <td>0.213573</td>\n",
       "      <td>0.311534</td>\n",
       "      <td>0.178801</td>\n",
       "      <td>0.260778</td>\n",
       "      <td>0.035314</td>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4900</td>\n",
       "      <td>0.222769</td>\n",
       "      <td>0.325123</td>\n",
       "      <td>0.165693</td>\n",
       "      <td>0.241778</td>\n",
       "      <td>0.044638</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>4897</td>\n",
       "      <td>0.024051</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>0.357534</td>\n",
       "      <td>0.521789</td>\n",
       "      <td>0.061521</td>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4902</td>\n",
       "      <td>0.152047</td>\n",
       "      <td>0.221978</td>\n",
       "      <td>0.233637</td>\n",
       "      <td>0.341042</td>\n",
       "      <td>0.051295</td>\n",
       "      <td>Brain_Cerebellum</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>4898</td>\n",
       "      <td>0.194933</td>\n",
       "      <td>0.284534</td>\n",
       "      <td>0.175424</td>\n",
       "      <td>0.255969</td>\n",
       "      <td>0.089141</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4902</td>\n",
       "      <td>0.236753</td>\n",
       "      <td>0.345643</td>\n",
       "      <td>0.153887</td>\n",
       "      <td>0.224624</td>\n",
       "      <td>0.039093</td>\n",
       "      <td>Brain_Frontal_Cortex_BA9</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4902</td>\n",
       "      <td>0.196699</td>\n",
       "      <td>0.287173</td>\n",
       "      <td>0.193582</td>\n",
       "      <td>0.282583</td>\n",
       "      <td>0.039963</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4894</td>\n",
       "      <td>0.238501</td>\n",
       "      <td>0.347976</td>\n",
       "      <td>0.152318</td>\n",
       "      <td>0.222195</td>\n",
       "      <td>0.039010</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4904</td>\n",
       "      <td>0.195124</td>\n",
       "      <td>0.284910</td>\n",
       "      <td>0.191641</td>\n",
       "      <td>0.279775</td>\n",
       "      <td>0.048550</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4892</td>\n",
       "      <td>0.212469</td>\n",
       "      <td>0.309992</td>\n",
       "      <td>0.172059</td>\n",
       "      <td>0.250980</td>\n",
       "      <td>0.054501</td>\n",
       "      <td>Brain_Putamen_basal_ganglia</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4862</td>\n",
       "      <td>0.234231</td>\n",
       "      <td>0.341017</td>\n",
       "      <td>0.159484</td>\n",
       "      <td>0.232160</td>\n",
       "      <td>0.033108</td>\n",
       "      <td>Brain_Spinal_cord_cervical_c-1</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>4388</td>\n",
       "      <td>0.253281</td>\n",
       "      <td>0.361726</td>\n",
       "      <td>0.135103</td>\n",
       "      <td>0.192892</td>\n",
       "      <td>0.056999</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    nsnps  PP.H0.abf  PP.H1.abf  PP.H2.abf  PP.H3.abf  PP.H4.abf  \\\n",
       "0    2015   0.636188   0.098066   0.218374   0.033648   0.013724   \n",
       "1    2022   0.644494   0.099806   0.209482   0.032427   0.013792   \n",
       "2    2034   0.013330   0.002080   0.825616   0.128778   0.030196   \n",
       "3    2032   0.060078   0.009348   0.779921   0.121331   0.029322   \n",
       "4    2035   0.000059   0.000009   0.839204   0.130955   0.029773   \n",
       "5    2035   0.459558   0.071636   0.389932   0.060765   0.018108   \n",
       "6    2028   0.105517   0.016404   0.735825   0.114366   0.027888   \n",
       "7    2035   0.469551   0.073282   0.379477   0.059206   0.018485   \n",
       "8    2025   0.375991   0.058281   0.471092   0.073000   0.021637   \n",
       "9    2035   0.183217   0.028599   0.660042   0.103003   0.025139   \n",
       "10   2027   0.045461   0.007065   0.792213   0.123079   0.032183   \n",
       "11   2018   0.146741   0.022651   0.695379   0.107309   0.027920   \n",
       "12   2006   0.653768   0.100076   0.201305   0.030801   0.014050   \n",
       "13   4859   0.239418   0.348457   0.152259   0.221563   0.038303   \n",
       "14   4892   0.213573   0.311534   0.178801   0.260778   0.035314   \n",
       "15   4900   0.222769   0.325123   0.165693   0.241778   0.044638   \n",
       "16   4897   0.024051   0.035104   0.357534   0.521789   0.061521   \n",
       "17   4902   0.152047   0.221978   0.233637   0.341042   0.051295   \n",
       "18   4898   0.194933   0.284534   0.175424   0.255969   0.089141   \n",
       "19   4902   0.236753   0.345643   0.153887   0.224624   0.039093   \n",
       "20   4902   0.196699   0.287173   0.193582   0.282583   0.039963   \n",
       "21   4894   0.238501   0.347976   0.152318   0.222195   0.039010   \n",
       "22   4904   0.195124   0.284910   0.191641   0.279775   0.048550   \n",
       "23   4892   0.212469   0.309992   0.172059   0.250980   0.054501   \n",
       "24   4862   0.234231   0.341017   0.159484   0.232160   0.033108   \n",
       "25   4388   0.253281   0.361726   0.135103   0.192892   0.056999   \n",
       "\n",
       "                                   Tissue index_variant  \n",
       "0                          Brain_Amygdala   rs141773145  \n",
       "1    Brain_Anterior_cingulate_cortex_BA24   rs141773145  \n",
       "2             Brain_Caudate_basal_ganglia   rs141773145  \n",
       "3             Brain_Cerebellar_Hemisphere   rs141773145  \n",
       "4                        Brain_Cerebellum   rs141773145  \n",
       "5                            Brain_Cortex   rs141773145  \n",
       "6                Brain_Frontal_Cortex_BA9   rs141773145  \n",
       "7                       Brain_Hippocampus   rs141773145  \n",
       "8                      Brain_Hypothalamus   rs141773145  \n",
       "9   Brain_Nucleus_accumbens_basal_ganglia   rs141773145  \n",
       "10            Brain_Putamen_basal_ganglia   rs141773145  \n",
       "11         Brain_Spinal_cord_cervical_c-1   rs141773145  \n",
       "12                 Brain_Substantia_nigra   rs141773145  \n",
       "13                         Brain_Amygdala    rs12860838  \n",
       "14   Brain_Anterior_cingulate_cortex_BA24    rs12860838  \n",
       "15            Brain_Caudate_basal_ganglia    rs12860838  \n",
       "16            Brain_Cerebellar_Hemisphere    rs12860838  \n",
       "17                       Brain_Cerebellum    rs12860838  \n",
       "18                           Brain_Cortex    rs12860838  \n",
       "19               Brain_Frontal_Cortex_BA9    rs12860838  \n",
       "20                      Brain_Hippocampus    rs12860838  \n",
       "21                     Brain_Hypothalamus    rs12860838  \n",
       "22  Brain_Nucleus_accumbens_basal_ganglia    rs12860838  \n",
       "23            Brain_Putamen_basal_ganglia    rs12860838  \n",
       "24         Brain_Spinal_cord_cervical_c-1    rs12860838  \n",
       "25                 Brain_Substantia_nigra    rs12860838  "
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "compiled_post = pd.read_csv(\"Analysis.COLOC_redo/females_only/coloc_posteriors_compiled_allIndexVars_allBrainTissues.txt\",sep=\"\\t\")\n",
    "compiled_post"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nsnps</th>\n",
       "      <th>PP.H0.abf</th>\n",
       "      <th>PP.H1.abf</th>\n",
       "      <th>PP.H2.abf</th>\n",
       "      <th>PP.H3.abf</th>\n",
       "      <th>PP.H4.abf</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>index_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2013</td>\n",
       "      <td>0.627853</td>\n",
       "      <td>0.105706</td>\n",
       "      <td>0.215384</td>\n",
       "      <td>0.036247</td>\n",
       "      <td>0.014810</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2022</td>\n",
       "      <td>0.635944</td>\n",
       "      <td>0.107563</td>\n",
       "      <td>0.206703</td>\n",
       "      <td>0.034947</td>\n",
       "      <td>0.014843</td>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2032</td>\n",
       "      <td>0.013149</td>\n",
       "      <td>0.002238</td>\n",
       "      <td>0.814372</td>\n",
       "      <td>0.138566</td>\n",
       "      <td>0.031676</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2030</td>\n",
       "      <td>0.059295</td>\n",
       "      <td>0.010071</td>\n",
       "      <td>0.769760</td>\n",
       "      <td>0.130705</td>\n",
       "      <td>0.030169</td>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2033</td>\n",
       "      <td>0.000058</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.827426</td>\n",
       "      <td>0.140841</td>\n",
       "      <td>0.031665</td>\n",
       "      <td>Brain_Cerebellum</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2033</td>\n",
       "      <td>0.453457</td>\n",
       "      <td>0.077141</td>\n",
       "      <td>0.384662</td>\n",
       "      <td>0.065419</td>\n",
       "      <td>0.019321</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2026</td>\n",
       "      <td>0.104216</td>\n",
       "      <td>0.017674</td>\n",
       "      <td>0.726726</td>\n",
       "      <td>0.123219</td>\n",
       "      <td>0.028165</td>\n",
       "      <td>Brain_Frontal_Cortex_BA9</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2033</td>\n",
       "      <td>0.463549</td>\n",
       "      <td>0.078907</td>\n",
       "      <td>0.374529</td>\n",
       "      <td>0.063734</td>\n",
       "      <td>0.019280</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2024</td>\n",
       "      <td>0.371126</td>\n",
       "      <td>0.062833</td>\n",
       "      <td>0.464955</td>\n",
       "      <td>0.078697</td>\n",
       "      <td>0.022388</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2033</td>\n",
       "      <td>0.180811</td>\n",
       "      <td>0.030785</td>\n",
       "      <td>0.651338</td>\n",
       "      <td>0.110870</td>\n",
       "      <td>0.026196</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2026</td>\n",
       "      <td>0.044825</td>\n",
       "      <td>0.007605</td>\n",
       "      <td>0.781116</td>\n",
       "      <td>0.132486</td>\n",
       "      <td>0.033968</td>\n",
       "      <td>Brain_Putamen_basal_ganglia</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2017</td>\n",
       "      <td>0.144973</td>\n",
       "      <td>0.024422</td>\n",
       "      <td>0.686984</td>\n",
       "      <td>0.115702</td>\n",
       "      <td>0.027918</td>\n",
       "      <td>Brain_Spinal_cord_cervical_c-1</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2005</td>\n",
       "      <td>0.644712</td>\n",
       "      <td>0.107909</td>\n",
       "      <td>0.198448</td>\n",
       "      <td>0.033199</td>\n",
       "      <td>0.015732</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>rs141773145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4253</td>\n",
       "      <td>0.189518</td>\n",
       "      <td>0.301124</td>\n",
       "      <td>0.180795</td>\n",
       "      <td>0.287223</td>\n",
       "      <td>0.041340</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>4225</td>\n",
       "      <td>0.105568</td>\n",
       "      <td>0.167030</td>\n",
       "      <td>0.268946</td>\n",
       "      <td>0.425492</td>\n",
       "      <td>0.032964</td>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4278</td>\n",
       "      <td>0.113175</td>\n",
       "      <td>0.180149</td>\n",
       "      <td>0.257502</td>\n",
       "      <td>0.409848</td>\n",
       "      <td>0.039326</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>4278</td>\n",
       "      <td>0.025137</td>\n",
       "      <td>0.040015</td>\n",
       "      <td>0.350065</td>\n",
       "      <td>0.557220</td>\n",
       "      <td>0.027562</td>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4273</td>\n",
       "      <td>0.001273</td>\n",
       "      <td>0.002027</td>\n",
       "      <td>0.368051</td>\n",
       "      <td>0.585689</td>\n",
       "      <td>0.042960</td>\n",
       "      <td>Brain_Cerebellum</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>4281</td>\n",
       "      <td>0.090069</td>\n",
       "      <td>0.143396</td>\n",
       "      <td>0.278152</td>\n",
       "      <td>0.442793</td>\n",
       "      <td>0.045590</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4257</td>\n",
       "      <td>0.018206</td>\n",
       "      <td>0.028942</td>\n",
       "      <td>0.357013</td>\n",
       "      <td>0.567510</td>\n",
       "      <td>0.028330</td>\n",
       "      <td>Brain_Frontal_Cortex_BA9</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4261</td>\n",
       "      <td>0.234623</td>\n",
       "      <td>0.372933</td>\n",
       "      <td>0.134796</td>\n",
       "      <td>0.214216</td>\n",
       "      <td>0.043432</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4241</td>\n",
       "      <td>0.151973</td>\n",
       "      <td>0.241302</td>\n",
       "      <td>0.215196</td>\n",
       "      <td>0.341637</td>\n",
       "      <td>0.049892</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4264</td>\n",
       "      <td>0.056853</td>\n",
       "      <td>0.090435</td>\n",
       "      <td>0.315661</td>\n",
       "      <td>0.502082</td>\n",
       "      <td>0.034969</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4250</td>\n",
       "      <td>0.063212</td>\n",
       "      <td>0.100421</td>\n",
       "      <td>0.311589</td>\n",
       "      <td>0.494977</td>\n",
       "      <td>0.029801</td>\n",
       "      <td>Brain_Putamen_basal_ganglia</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4255</td>\n",
       "      <td>0.014745</td>\n",
       "      <td>0.023427</td>\n",
       "      <td>0.361919</td>\n",
       "      <td>0.574999</td>\n",
       "      <td>0.024910</td>\n",
       "      <td>Brain_Spinal_cord_cervical_c-1</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>4249</td>\n",
       "      <td>0.165907</td>\n",
       "      <td>0.263503</td>\n",
       "      <td>0.201854</td>\n",
       "      <td>0.320548</td>\n",
       "      <td>0.048187</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>rs6648060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>602</td>\n",
       "      <td>0.897450</td>\n",
       "      <td>0.044088</td>\n",
       "      <td>0.051734</td>\n",
       "      <td>0.002537</td>\n",
       "      <td>0.004192</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>226</td>\n",
       "      <td>0.953211</td>\n",
       "      <td>0.022729</td>\n",
       "      <td>0.021426</td>\n",
       "      <td>0.000509</td>\n",
       "      <td>0.002125</td>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>607</td>\n",
       "      <td>0.892366</td>\n",
       "      <td>0.044203</td>\n",
       "      <td>0.056172</td>\n",
       "      <td>0.002778</td>\n",
       "      <td>0.004481</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>226</td>\n",
       "      <td>0.954257</td>\n",
       "      <td>0.022754</td>\n",
       "      <td>0.020382</td>\n",
       "      <td>0.000484</td>\n",
       "      <td>0.002123</td>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>227</td>\n",
       "      <td>0.953399</td>\n",
       "      <td>0.022833</td>\n",
       "      <td>0.021018</td>\n",
       "      <td>0.000501</td>\n",
       "      <td>0.002249</td>\n",
       "      <td>Brain_Cerebellum</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>227</td>\n",
       "      <td>0.952849</td>\n",
       "      <td>0.022820</td>\n",
       "      <td>0.021362</td>\n",
       "      <td>0.000509</td>\n",
       "      <td>0.002459</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>227</td>\n",
       "      <td>0.954092</td>\n",
       "      <td>0.022850</td>\n",
       "      <td>0.020428</td>\n",
       "      <td>0.000487</td>\n",
       "      <td>0.002143</td>\n",
       "      <td>Brain_Frontal_Cortex_BA9</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>606</td>\n",
       "      <td>0.882078</td>\n",
       "      <td>0.043626</td>\n",
       "      <td>0.066051</td>\n",
       "      <td>0.003262</td>\n",
       "      <td>0.004983</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>606</td>\n",
       "      <td>0.890482</td>\n",
       "      <td>0.044043</td>\n",
       "      <td>0.058087</td>\n",
       "      <td>0.002868</td>\n",
       "      <td>0.004520</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>608</td>\n",
       "      <td>0.886950</td>\n",
       "      <td>0.044028</td>\n",
       "      <td>0.061416</td>\n",
       "      <td>0.003044</td>\n",
       "      <td>0.004563</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>606</td>\n",
       "      <td>0.893779</td>\n",
       "      <td>0.044137</td>\n",
       "      <td>0.054873</td>\n",
       "      <td>0.002705</td>\n",
       "      <td>0.004507</td>\n",
       "      <td>Brain_Putamen_basal_ganglia</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1379</td>\n",
       "      <td>0.803904</td>\n",
       "      <td>0.074663</td>\n",
       "      <td>0.104475</td>\n",
       "      <td>0.009696</td>\n",
       "      <td>0.007262</td>\n",
       "      <td>Brain_Spinal_cord_cervical_c-1</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>220</td>\n",
       "      <td>0.955577</td>\n",
       "      <td>0.022178</td>\n",
       "      <td>0.019694</td>\n",
       "      <td>0.000455</td>\n",
       "      <td>0.002096</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>rs141193614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>4857</td>\n",
       "      <td>0.163112</td>\n",
       "      <td>0.422067</td>\n",
       "      <td>0.103697</td>\n",
       "      <td>0.268283</td>\n",
       "      <td>0.042841</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>4890</td>\n",
       "      <td>0.145576</td>\n",
       "      <td>0.377141</td>\n",
       "      <td>0.121845</td>\n",
       "      <td>0.315622</td>\n",
       "      <td>0.039816</td>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>4898</td>\n",
       "      <td>0.151449</td>\n",
       "      <td>0.392481</td>\n",
       "      <td>0.112615</td>\n",
       "      <td>0.291791</td>\n",
       "      <td>0.051665</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>4895</td>\n",
       "      <td>0.016590</td>\n",
       "      <td>0.042994</td>\n",
       "      <td>0.246622</td>\n",
       "      <td>0.639064</td>\n",
       "      <td>0.054730</td>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>4901</td>\n",
       "      <td>0.103407</td>\n",
       "      <td>0.268032</td>\n",
       "      <td>0.158884</td>\n",
       "      <td>0.411772</td>\n",
       "      <td>0.057905</td>\n",
       "      <td>Brain_Cerebellum</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>4896</td>\n",
       "      <td>0.128304</td>\n",
       "      <td>0.332517</td>\n",
       "      <td>0.115438</td>\n",
       "      <td>0.299047</td>\n",
       "      <td>0.124694</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>4900</td>\n",
       "      <td>0.160778</td>\n",
       "      <td>0.416725</td>\n",
       "      <td>0.104472</td>\n",
       "      <td>0.270735</td>\n",
       "      <td>0.047290</td>\n",
       "      <td>Brain_Frontal_Cortex_BA9</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>4901</td>\n",
       "      <td>0.134750</td>\n",
       "      <td>0.349286</td>\n",
       "      <td>0.132601</td>\n",
       "      <td>0.343676</td>\n",
       "      <td>0.039686</td>\n",
       "      <td>Brain_Hippocampus</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>4892</td>\n",
       "      <td>0.161588</td>\n",
       "      <td>0.418683</td>\n",
       "      <td>0.103165</td>\n",
       "      <td>0.267257</td>\n",
       "      <td>0.049307</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>4902</td>\n",
       "      <td>0.132324</td>\n",
       "      <td>0.343007</td>\n",
       "      <td>0.129935</td>\n",
       "      <td>0.336757</td>\n",
       "      <td>0.057977</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>4890</td>\n",
       "      <td>0.144551</td>\n",
       "      <td>0.374537</td>\n",
       "      <td>0.117030</td>\n",
       "      <td>0.303167</td>\n",
       "      <td>0.060714</td>\n",
       "      <td>Brain_Putamen_basal_ganglia</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>4860</td>\n",
       "      <td>0.159514</td>\n",
       "      <td>0.412767</td>\n",
       "      <td>0.108578</td>\n",
       "      <td>0.280926</td>\n",
       "      <td>0.038215</td>\n",
       "      <td>Brain_Spinal_cord_cervical_c-1</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>4386</td>\n",
       "      <td>0.170774</td>\n",
       "      <td>0.437420</td>\n",
       "      <td>0.091058</td>\n",
       "      <td>0.233168</td>\n",
       "      <td>0.067579</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>rs12860838</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    nsnps  PP.H0.abf  PP.H1.abf  PP.H2.abf  PP.H3.abf  PP.H4.abf  \\\n",
       "0    2013   0.627853   0.105706   0.215384   0.036247   0.014810   \n",
       "1    2022   0.635944   0.107563   0.206703   0.034947   0.014843   \n",
       "2    2032   0.013149   0.002238   0.814372   0.138566   0.031676   \n",
       "3    2030   0.059295   0.010071   0.769760   0.130705   0.030169   \n",
       "4    2033   0.000058   0.000010   0.827426   0.140841   0.031665   \n",
       "5    2033   0.453457   0.077141   0.384662   0.065419   0.019321   \n",
       "6    2026   0.104216   0.017674   0.726726   0.123219   0.028165   \n",
       "7    2033   0.463549   0.078907   0.374529   0.063734   0.019280   \n",
       "8    2024   0.371126   0.062833   0.464955   0.078697   0.022388   \n",
       "9    2033   0.180811   0.030785   0.651338   0.110870   0.026196   \n",
       "10   2026   0.044825   0.007605   0.781116   0.132486   0.033968   \n",
       "11   2017   0.144973   0.024422   0.686984   0.115702   0.027918   \n",
       "12   2005   0.644712   0.107909   0.198448   0.033199   0.015732   \n",
       "13   4253   0.189518   0.301124   0.180795   0.287223   0.041340   \n",
       "14   4225   0.105568   0.167030   0.268946   0.425492   0.032964   \n",
       "15   4278   0.113175   0.180149   0.257502   0.409848   0.039326   \n",
       "16   4278   0.025137   0.040015   0.350065   0.557220   0.027562   \n",
       "17   4273   0.001273   0.002027   0.368051   0.585689   0.042960   \n",
       "18   4281   0.090069   0.143396   0.278152   0.442793   0.045590   \n",
       "19   4257   0.018206   0.028942   0.357013   0.567510   0.028330   \n",
       "20   4261   0.234623   0.372933   0.134796   0.214216   0.043432   \n",
       "21   4241   0.151973   0.241302   0.215196   0.341637   0.049892   \n",
       "22   4264   0.056853   0.090435   0.315661   0.502082   0.034969   \n",
       "23   4250   0.063212   0.100421   0.311589   0.494977   0.029801   \n",
       "24   4255   0.014745   0.023427   0.361919   0.574999   0.024910   \n",
       "25   4249   0.165907   0.263503   0.201854   0.320548   0.048187   \n",
       "26    602   0.897450   0.044088   0.051734   0.002537   0.004192   \n",
       "27    226   0.953211   0.022729   0.021426   0.000509   0.002125   \n",
       "28    607   0.892366   0.044203   0.056172   0.002778   0.004481   \n",
       "29    226   0.954257   0.022754   0.020382   0.000484   0.002123   \n",
       "30    227   0.953399   0.022833   0.021018   0.000501   0.002249   \n",
       "31    227   0.952849   0.022820   0.021362   0.000509   0.002459   \n",
       "32    227   0.954092   0.022850   0.020428   0.000487   0.002143   \n",
       "33    606   0.882078   0.043626   0.066051   0.003262   0.004983   \n",
       "34    606   0.890482   0.044043   0.058087   0.002868   0.004520   \n",
       "35    608   0.886950   0.044028   0.061416   0.003044   0.004563   \n",
       "36    606   0.893779   0.044137   0.054873   0.002705   0.004507   \n",
       "37   1379   0.803904   0.074663   0.104475   0.009696   0.007262   \n",
       "38    220   0.955577   0.022178   0.019694   0.000455   0.002096   \n",
       "39   4857   0.163112   0.422067   0.103697   0.268283   0.042841   \n",
       "40   4890   0.145576   0.377141   0.121845   0.315622   0.039816   \n",
       "41   4898   0.151449   0.392481   0.112615   0.291791   0.051665   \n",
       "42   4895   0.016590   0.042994   0.246622   0.639064   0.054730   \n",
       "43   4901   0.103407   0.268032   0.158884   0.411772   0.057905   \n",
       "44   4896   0.128304   0.332517   0.115438   0.299047   0.124694   \n",
       "45   4900   0.160778   0.416725   0.104472   0.270735   0.047290   \n",
       "46   4901   0.134750   0.349286   0.132601   0.343676   0.039686   \n",
       "47   4892   0.161588   0.418683   0.103165   0.267257   0.049307   \n",
       "48   4902   0.132324   0.343007   0.129935   0.336757   0.057977   \n",
       "49   4890   0.144551   0.374537   0.117030   0.303167   0.060714   \n",
       "50   4860   0.159514   0.412767   0.108578   0.280926   0.038215   \n",
       "51   4386   0.170774   0.437420   0.091058   0.233168   0.067579   \n",
       "\n",
       "                                   Tissue index_variant  \n",
       "0                          Brain_Amygdala   rs141773145  \n",
       "1    Brain_Anterior_cingulate_cortex_BA24   rs141773145  \n",
       "2             Brain_Caudate_basal_ganglia   rs141773145  \n",
       "3             Brain_Cerebellar_Hemisphere   rs141773145  \n",
       "4                        Brain_Cerebellum   rs141773145  \n",
       "5                            Brain_Cortex   rs141773145  \n",
       "6                Brain_Frontal_Cortex_BA9   rs141773145  \n",
       "7                       Brain_Hippocampus   rs141773145  \n",
       "8                      Brain_Hypothalamus   rs141773145  \n",
       "9   Brain_Nucleus_accumbens_basal_ganglia   rs141773145  \n",
       "10            Brain_Putamen_basal_ganglia   rs141773145  \n",
       "11         Brain_Spinal_cord_cervical_c-1   rs141773145  \n",
       "12                 Brain_Substantia_nigra   rs141773145  \n",
       "13                         Brain_Amygdala     rs6648060  \n",
       "14   Brain_Anterior_cingulate_cortex_BA24     rs6648060  \n",
       "15            Brain_Caudate_basal_ganglia     rs6648060  \n",
       "16            Brain_Cerebellar_Hemisphere     rs6648060  \n",
       "17                       Brain_Cerebellum     rs6648060  \n",
       "18                           Brain_Cortex     rs6648060  \n",
       "19               Brain_Frontal_Cortex_BA9     rs6648060  \n",
       "20                      Brain_Hippocampus     rs6648060  \n",
       "21                     Brain_Hypothalamus     rs6648060  \n",
       "22  Brain_Nucleus_accumbens_basal_ganglia     rs6648060  \n",
       "23            Brain_Putamen_basal_ganglia     rs6648060  \n",
       "24         Brain_Spinal_cord_cervical_c-1     rs6648060  \n",
       "25                 Brain_Substantia_nigra     rs6648060  \n",
       "26                         Brain_Amygdala   rs141193614  \n",
       "27   Brain_Anterior_cingulate_cortex_BA24   rs141193614  \n",
       "28            Brain_Caudate_basal_ganglia   rs141193614  \n",
       "29            Brain_Cerebellar_Hemisphere   rs141193614  \n",
       "30                       Brain_Cerebellum   rs141193614  \n",
       "31                           Brain_Cortex   rs141193614  \n",
       "32               Brain_Frontal_Cortex_BA9   rs141193614  \n",
       "33                      Brain_Hippocampus   rs141193614  \n",
       "34                     Brain_Hypothalamus   rs141193614  \n",
       "35  Brain_Nucleus_accumbens_basal_ganglia   rs141193614  \n",
       "36            Brain_Putamen_basal_ganglia   rs141193614  \n",
       "37         Brain_Spinal_cord_cervical_c-1   rs141193614  \n",
       "38                 Brain_Substantia_nigra   rs141193614  \n",
       "39                         Brain_Amygdala    rs12860838  \n",
       "40   Brain_Anterior_cingulate_cortex_BA24    rs12860838  \n",
       "41            Brain_Caudate_basal_ganglia    rs12860838  \n",
       "42            Brain_Cerebellar_Hemisphere    rs12860838  \n",
       "43                       Brain_Cerebellum    rs12860838  \n",
       "44                           Brain_Cortex    rs12860838  \n",
       "45               Brain_Frontal_Cortex_BA9    rs12860838  \n",
       "46                      Brain_Hippocampus    rs12860838  \n",
       "47                     Brain_Hypothalamus    rs12860838  \n",
       "48  Brain_Nucleus_accumbens_basal_ganglia    rs12860838  \n",
       "49            Brain_Putamen_basal_ganglia    rs12860838  \n",
       "50         Brain_Spinal_cord_cervical_c-1    rs12860838  \n",
       "51                 Brain_Substantia_nigra    rs12860838  "
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "compiled_post_cond = pd.read_csv(\"Analysis.COLOC_redo/females_only_cond_ApoE4/coloc_posteriors_compiled_allIndexVars_allBrainTissues.txt\",sep=\"\\t\")\n",
    "compiled_post_cond"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.10",
   "language": "python",
   "name": "py3.10"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "388.508px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "toc-autonumbering": true,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
